# Annotate Transmission Summary

These scripts enable the annotation and subsequent filtering of transmission summary files. There are two types of transmission summary files: flat (one line per sample-variant) and family aggregation (one line per variant-family). Analysis-based filtering options described here: https://docs.google.com/spreadsheets/d/1d28ojG4sLbMGYldkZrDMTaLf2rylM4_cCL7uvEEzZPo/edit?usp=sharing.

## Guideline
1. Main functions of these scripts
2. Main scripts
3. Overview of analysis based filtering script
4. Additional support scripts
5. Instructions for adding a new project
6. Hints for adding a new column

## 1. Main functions of these scripts:
1. **MAKE SHARDS**: To improve compute performance, split large transmission summary files into smaller shards
2. **ANNOTATION**: Annotate flat and family aggregation transmission summary files with critical information (e.g., Genotype , GIAB region, HGNC name)
3. **ANALYSIS BASED FILTERING OF A FILTERED FILE**: Not parallelized. Only recommended for filtering a smaller pre-filtered file (no shards or per chr features built in)
4. **ANALYSIS BASED FILTERING**: Filter annotated flat and family aggregation transmission summary files from shards and recombine shards into one file. 

## 2. Main scripts:
1. **MAKE SHARDS**: To improve compute performance, split large transmission summary files into smaller shards
    1. /ifs/collabs/geschwind/bmap_scripts/process_vcf_line_by_line/split_and_gzip.py #source code to split
    2. qsub_split_gz_ms2.sh #qsub submission code example. User must make a project-specific version of a script. 

2. **ANNOTATION**: Annotate flat and family aggregation transmission summary files with critical information (e.g., Genotype, GIAB region, HGNC name)
    1. annotate_flat_transmission_summary.py #annotates flat files
    2. annotate_family_aggregation_transmission_summary.py #annotates family aggregation files
    3. qsub_shards_iHART.sh #qsub submission code (flat & family) example. User must make a project-specific version of a script. 

3. **ANALYSIS BASED FILTERING OF A FILTERED FILE**: Not parallelized. Only recommended for filtering a smaller pre-filtered file (no shards or per chr features built in)
    1. filter_flat_annotated_transmission_summary.py
    2. filter_family_aggregation_annotated_transmission_summary.py

4. **ANALYSIS BASED FILTERING**: Filter annotated flat and family aggregation transmission summary files from shards and recombine shards into one file. <br/>
   1. filter_annotated_transmission_summary_qsub.sh #Most efficent for first large filtering step of main projects on bmap.
   2. combine_and_clean.sh #source code at the end of filter_annotated_transmission_summary_qsub.sh to give you a single filtered output file.
   3. Filtering_Wrapper_Script.sh #Most efficent for first large filtering step of single file or multiple input files. <br/>
   4. combine_and_clean_script.sh #source code within Filtering_Wrapper_Script.sh to give you requested output type. <br/>

## 3. Overview of analysis based filtering scripts:
Two Input Types
-----------
  1. Single Transmission Summary File
  2. Multiple Transmission Summary Files within a single directory
  
Required Arguments:
-----------
   ##### Single Transmission Summary File Arguments
   1. **--input_db**: absolute path to input transmission summary file
   2. **--output_dir**: absolute path to desired output directory
   3. **--input_type**: specify transmission summary input format as either: (per_chr or multi_chr, see details below) <br/>
   4. **--flat_or_family**: specify whether input is a flat or family aggregate transmission summary file: family or flat
   
   ##### Multiple Transmission Summary Files within a single directory
   1. **--input_db**: absolute path to directory with multiple transmission summary files
   2. **--input_prefix**: string preceding chromosome specification (and shard number if present)
     
      * Example 1: <br/>
      Filename: Transmission_Summary_chr1_annotation.db <br/>
      Prefix: Transmission_Summary_chr
      
      * Example 2: <br/>
      Filename: Transmission_Summary_chr1_00000_annotation.db <br/>
      Prefix: Transmission_Summary_chr
    
   3. **--input_suffix**: string following chromosome specification (and shard number if present) 
     
      * Example 1: <br/>
      Filename: Transmission_Summary_chr1_annotation.db <br/>
      Prefix: _annotation.db
      
      * Example 2: <br/>
      Filename: Transmission_Summary_chr1_00000_annotation.db <br/>
      Prefix: _annotation.db
      
   3. **--output_dir**: absolute path to desired output directory
   3. **--input_type**: specify transmission summary input format as either: (per_chr or chr_shards, see details below) <br/>
   4. **--flat_or_family**: specify whether input is a flat or family aggregate transmission summary file: flat or family
   
Optional Arguments:
-----------
1. **--comb_and_clean**: specify output format as either: merge_all_chr (Default), merge_by_chr, no_comb_clean
2. **--no_qsub**: specify whether qsub should not be used: True, False (Default)
3. **--log_dir**: specify desired log directory (Default: output_dir/Logs)
4. **--exclude_chr**: specify if specific chromosomes should be excluded from filtering (Default: None)

Input and Output Formats:
-----------
The wrapper script accepts transmission summary files in three formats: per_chr, multi_chr, and chr_shards (see below for details)
![Transmission Summary Input formats and input_type](https://imgur.com/jlXZgNH.png)

Input in multi_chr format are preprocessed by the script and divided into per chromosome shards in a directory named "filtered_flat_per_chr_shards" or "filtered_family_per_chr_shards" within the user specified output directory (see below).
![Preprocessing Step](https://imgur.com/zrk71Hm.png)

The wrapper script also allows the user to specify their desired filtering output format by using the --comb_and_clean argument to select: merge_all_chr (Default), merge_by_chr, or no_comb_clean (see details below).
![Combine and Clean Options](https://imgur.com/eRqXKbU.png)

Example Commands:
-----------
Filtering is most commonly conducted on single, per_chr transmission summary files and on multiple, chr_shard files. Example commands for the filtering wrapper script are provided first for these most common filtering cases, followed by the less common single, multi_chr and multiple, per_chr use cases.

#### Individual File Example (per_chr)
---
**General Command**:
```
bash Filtering_Wrapper_Script.sh --input_db <input_db> --output_dir <output_dir> --input_type per_chr --flat_or_family <flat_or_family (e.g. flat, family)> [--comb_and_clean <comb_and_clean> (e.g. merge_all_chr, merge_by_chr, or no_comb_clean)] [--no_qsub <no_qsub (e.g. True, False)>] [--log_dir <log_dir>] [--exclude_chr <exclude_chr>]
```

**Input File**: TS_filtering_examples/single_file_per_chr/Transmission_Summary_Flat_chr1_annotated_for_analysis.db
```
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
1  10492 NA C  T  NA NA NA NA 0.99208  10.62 NA NA NA NA upstream_gene_variant   DDX11L1  ABCDEF   ABCDEF101   1  uncertain   PASS;ABHet=0.734;ABHom=0.953;AC=1095;AF=0.119;AN=9170;BaseQRankSum=-9.500e-01;DP=151995;FS=1.412;GC=95.24;GQ_MEAN=56.05;GQ_STDDEV=42.80;InbreedingCoeff=-0.1477;MLEAC=1158;MLEAF=0.126;MQ=53.11;MQ0=0;MQRankSum=0.043;NCC=25;NDA=1;OND=5.382e-03;QD=3.83;ReadPosRankSum=0.00;SOR=0.445;VQSLOD=-5.762e+00;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=NONE,DDX11L1;GeneDetail=NONE,dist=1377;genomicSuperDups=0.993729,chr9:10843;CADD=0.992080;CADD_Phred=10.62;FATHMM_c=0.73423;FATHMM_nc=0.05648;GWAVA_region=0.4;GWAVA_tss=0.26;GWAVA_unmatched=182;CSQ=T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000456328|processed_transcript||||||||||rs55998931|1377|1|SNV|HGNC|37102|YES|||||||||||||||||||||||||||||||||||||||||  0/1   0.330:3,6:9:67:.:.:162,0,67   GT:AB:AD:DP:GQ:PGT:PID:PL  0.0811632   0.120671284 0  0.004794885 0.527704485 0.687830688 NA 1  DDX11L1  ambiguous_previous   NA NA NA NA NA NA NA NA NA NA NA NA NA 0.147540984 0.00192864  0.096339114 8.12E-02
1  10492 NA C  T  NA NA NA NA 0.99208  10.62 NA NA NA NA upstream_gene_variant   DDX11L1  ABCDEF   ABCDEF102   1  uncertain   PASS;ABHet=0.734;ABHom=0.953;AC=1095;AF=0.119;AN=9170;BaseQRankSum=-9.500e-01;DP=151995;FS=1.412;GC=95.24;GQ_MEAN=56.05;GQ_STDDEV=42.80;InbreedingCoeff=-0.1477;MLEAC=1158;MLEAF=0.126;MQ=53.11;MQ0=0;MQRankSum=0.043;NCC=25;NDA=1;OND=5.382e-03;QD=3.83;ReadPosRankSum=0.00;SOR=0.445;VQSLOD=-5.762e+00;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=NONE,DDX11L1;GeneDetail=NONE,dist=1377;genomicSuperDups=0.993729,chr9:10843;CADD=0.992080;CADD_Phred=10.62;FATHMM_c=0.73423;FATHMM_nc=0.05648;GWAVA_region=0.4;GWAVA_tss=0.26;GWAVA_unmatched=182;CSQ=T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000456328|processed_transcript||||||||||rs55998931|1377|1|SNV|HGNC|37102|YES|||||||||||||||||||||||||||||||||||||||||  0/0   .:23,0:23:0:.:.:0,0,36  GT:AB:AD:DP:GQ:PGT:PID:PL  0.0811632   0.120671284 0  0.004794885 0.527704485 0.687830688 NA 1  DDX11L1  ambiguous_previous   NA NA NA NA NA NA NA NA NA NA NA NA 0.4   0.147540984 0.00192864  0.096339114 8.12E-02
1  10492 NA C  T  NA NA NA NA 0.99208  10.62 NA NA NA NA upstream_gene_variant   DDX11L1  ABCDEF   ABCDEF103   0  uncertain   PASS;ABHet=0.734;ABHom=0.953;AC=1095;AF=0.119;AN=9170;BaseQRankSum=-9.500e-01;DP=151995;FS=1.412;GC=95.24;GQ_MEAN=56.05;GQ_STDDEV=42.80;InbreedingCoeff=-0.1477;MLEAC=1158;MLEAF=0.126;MQ=53.11;MQ0=0;MQRankSum=0.043;NCC=25;NDA=1;OND=5.382e-03;QD=3.83;ReadPosRankSum=0.00;SOR=0.445;VQSLOD=-5.762e+00;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=NONE,DDX11L1;GeneDetail=NONE,dist=1377;genomicSuperDups=0.993729,chr9:10843;CADD=0.992080;CADD_Phred=10.62;FATHMM_c=0.73423;FATHMM_nc=0.05648;GWAVA_region=0.4;GWAVA_tss=0.26;GWAVA_unmatched=182;CSQ=T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000456328|processed_transcript||||||||||rs55998931|1377|1|SNV|HGNC|37102|YES|||||||||||||||||||||||||||||||||||||||||  0/0   .:37,4:41:14:.:.:0,14,1156 GT:AB:AD:DP:GQ:PGT:PID:PL  0.0811632   0.120671284 0  0.004794885 0.527704485 0.687830688 NA 1  DDX11L1  ambiguous_previous   NA NA NA NA NA NA NA NA NA NA NA NA 0.3   0.147540984 0.00192864  0.096339114 8.12E-02
1  10492 NA C  T  NA NA NA NA 0.99208  10.62 NA NA NA NA upstream_gene_variant   DDX11L1  ABCDEF   ABCDEF104   1  uncertain   PASS;ABHet=0.734;ABHom=0.953;AC=1095;AF=0.119;AN=9170;BaseQRankSum=-9.500e-01;DP=151995;FS=1.412;GC=95.24;GQ_MEAN=56.05;GQ_STDDEV=42.80;InbreedingCoeff=-0.1477;MLEAC=1158;MLEAF=0.126;MQ=53.11;MQ0=0;MQRankSum=0.043;NCC=25;NDA=1;OND=5.382e-03;QD=3.83;ReadPosRankSum=0.00;SOR=0.445;VQSLOD=-5.762e+00;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=NONE,DDX11L1;GeneDetail=NONE,dist=1377;genomicSuperDups=0.993729,chr9:10843;CADD=0.992080;CADD_Phred=10.62;FATHMM_c=0.73423;FATHMM_nc=0.05648;GWAVA_region=0.4;GWAVA_tss=0.26;GWAVA_unmatched=182;CSQ=T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000456328|processed_transcript||||||||||rs55998931|1377|1|SNV|HGNC|37102|YES|||||||||||||||||||||||||||||||||||||||||  0/1   0.710:10,4:14:68:.:.:68,0,239 GT:AB:AD:DP:GQ:PGT:PID:PL  0.0811632   0.120671284 0  0.004794885 0.527704485 0.687830688 NA 1  DDX11L1  ambiguous_previous   NA NA NA NA NA NA NA NA NA NA NA NA 0.1   0.147540984 0.00192864  0.096339114 8.12E-02
```

**Command**:
```
bash /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/Filtering_Wrapper_Script.sh --input_db /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/single_file_per_chr/Transmission_Summary_Flat_chr1_annotated_for_analysis.db --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/single_file_per_chr --input_type per_chr --flat_or_family flat --ARC_score ge0.4,NA
```

**Output File**: TS_filtering_examples/single_file_per_chr/Transmission_Summary_Flat_chr1_annotated_for_analysis_filtered.db
```
#bmap command: /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/filter_flat_annotated_transmission_summary.py --comb_and_clean no_comb_clean --input_db /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/single_file_per_chr/Transmission_Summary_Flat_chr1_annotated_for_analysis.db --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/single_file_per_chr --input_type per_chr --flat_or_family flat --ARC_score ge0.4,NA
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
1  10492 NA C  T  NA NA NA NA 0.99208  10.62 NA NA NA NA upstream_gene_variant   DDX11L1  ABCDEF   ABCDEF101   1  uncertain   PASS;ABHet=0.734;ABHom=0.953;AC=1095;AF=0.119;AN=9170;BaseQRankSum=-9.500e-01;DP=151995;FS=1.412;GC=95.24;GQ_MEAN=56.05;GQ_STDDEV=42.80;InbreedingCoeff=-0.1477;MLEAC=1158;MLEAF=0.126;MQ=53.11;MQ0=0;MQRankSum=0.043;NCC=25;NDA=1;OND=5.382e-03;QD=3.83;ReadPosRankSum=0.00;SOR=0.445;VQSLOD=-5.762e+00;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=NONE,DDX11L1;GeneDetail=NONE,dist=1377;genomicSuperDups=0.993729,chr9:10843;CADD=0.992080;CADD_Phred=10.62;FATHMM_c=0.73423;FATHMM_nc=0.05648;GWAVA_region=0.4;GWAVA_tss=0.26;GWAVA_unmatched=182;CSQ=T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000456328|processed_transcript||||||||||rs55998931|1377|1|SNV|HGNC|37102|YES|||||||||||||||||||||||||||||||||||||||||  0/1   0.330:3,6:9:67:.:.:162,0,67   GT:AB:AD:DP:GQ:PGT:PID:PL  0.0811632   0.120671284 0  0.004794885 0.527704485 0.687830688 NA 1  DDX11L1  ambiguous_previous   NA NA NA NA NA NA NA NA NA NA NA NA NA 0.147540984 0.00192864  0.096339114 8.12E-02
1  10492 NA C  T  NA NA NA NA 0.99208  10.62 NA NA NA NA upstream_gene_variant   DDX11L1  ABCDEF   ABCDEF102   1  uncertain   PASS;ABHet=0.734;ABHom=0.953;AC=1095;AF=0.119;AN=9170;BaseQRankSum=-9.500e-01;DP=151995;FS=1.412;GC=95.24;GQ_MEAN=56.05;GQ_STDDEV=42.80;InbreedingCoeff=-0.1477;MLEAC=1158;MLEAF=0.126;MQ=53.11;MQ0=0;MQRankSum=0.043;NCC=25;NDA=1;OND=5.382e-03;QD=3.83;ReadPosRankSum=0.00;SOR=0.445;VQSLOD=-5.762e+00;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=NONE,DDX11L1;GeneDetail=NONE,dist=1377;genomicSuperDups=0.993729,chr9:10843;CADD=0.992080;CADD_Phred=10.62;FATHMM_c=0.73423;FATHMM_nc=0.05648;GWAVA_region=0.4;GWAVA_tss=0.26;GWAVA_unmatched=182;CSQ=T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000456328|processed_transcript||||||||||rs55998931|1377|1|SNV|HGNC|37102|YES|||||||||||||||||||||||||||||||||||||||||  0/0   .:23,0:23:0:.:.:0,0,36  GT:AB:AD:DP:GQ:PGT:PID:PL  0.0811632   0.120671284 0  0.004794885 0.527704485 0.687830688 NA 1  DDX11L1  ambiguous_previous   NA NA NA NA NA NA NA NA NA NA NA NA 0.4   0.147540984 0.00192864  0.096339114 8.12E-02
```

#### Multiple Files Example (chr_shards)
---

**Input Files Within Subdirectory of input_dir**:

Chromosome 1 Input File (Shard 1): TS_filtering_examples/multiple_file_chr_shards/subdir_input/1/Transmission_Summary_Flat_chr1_00000_annotated_for_analysis.db
```
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
1  10492 NA C  T  NA NA NA NA 0.99208  10.62 NA NA NA NA upstream_gene_variant   DDX11L1  ABCDEF   ABCDEF100   1  uncertain   PASS;ABHet=0.734;ABHom=0.953;AC=1095;AF=0.119;AN=9170;BaseQRankSum=-9.500e-01;DP=151995;FS=1.412;GC=95.24;GQ_MEAN=56.05;GQ_STDDEV=42.80;InbreedingCoeff=-0.1477;MLEAC=1158;MLEAF=0.126;MQ=53.11;MQ0=0;MQRankSum=0.043;NCC=25;NDA=1;OND=5.382e-03;QD=3.83;ReadPosRankSum=0.00;SOR=0.445;VQSLOD=-5.762e+00;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=NONE,DDX11L1;GeneDetail=NONE,dist=1377;genomicSuperDups=0.993729,chr9:10843;CADD=0.992080;CADD_Phred=10.62;FATHMM_c=0.73423;FATHMM_nc=0.05648;GWAVA_region=0.4;GWAVA_tss=0.26;GWAVA_unmatched=182;CSQ=T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000456328|processed_transcript||||||||||rs55998931|1377|1|SNV|HGNC|37102|YES|||||||||||||||||||||||||||||||||||||||||  0/1   0.330:3,6:9:67:.:.:162,0,67   GT:AB:AD:DP:GQ:PGT:PID:PL  0.0811632   0.120671284 0  0.004794885 0.527704485 0.687830688 NA 1  DDX11L1  ambiguous_previous   NA NA NA NA NA NA NA NA NA NA NA NA NA 0.147540984 0.00192864  0.096339114 8.12E-02
1  10492 NA C  T  NA NA NA NA 0.99208  10.62 NA NA NA NA upstream_gene_variant   DDX11L1  ABCDEF   ABCDEF101   1  uncertain   PASS;ABHet=0.734;ABHom=0.953;AC=1095;AF=0.119;AN=9170;BaseQRankSum=-9.500e-01;DP=151995;FS=1.412;GC=95.24;GQ_MEAN=56.05;GQ_STDDEV=42.80;InbreedingCoeff=-0.1477;MLEAC=1158;MLEAF=0.126;MQ=53.11;MQ0=0;MQRankSum=0.043;NCC=25;NDA=1;OND=5.382e-03;QD=3.83;ReadPosRankSum=0.00;SOR=0.445;VQSLOD=-5.762e+00;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=NONE,DDX11L1;GeneDetail=NONE,dist=1377;genomicSuperDups=0.993729,chr9:10843;CADD=0.992080;CADD_Phred=10.62;FATHMM_c=0.73423;FATHMM_nc=0.05648;GWAVA_region=0.4;GWAVA_tss=0.26;GWAVA_unmatched=182;CSQ=T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000456328|processed_transcript||||||||||rs55998931|1377|1|SNV|HGNC|37102|YES|||||||||||||||||||||||||||||||||||||||||  0/0   .:23,0:23:0:.:.:0,0,36  GT:AB:AD:DP:GQ:PGT:PID:PL  0.0811632   0.120671284 0  0.004794885 0.527704485 0.687830688 NA 1  DDX11L1  ambiguous_previous   NA NA NA NA NA NA NA NA NA NA NA NA NA 0.147540984 0.00192864  0.096339114 8.12E-02
```
Chromosome 1 Input File (Shard 2): TS_filtering_examples/multiple_file_chr_shards/subdir_input/1/Transmission_Summary_Flat_chr1_00001_annotated_for_analysis.db
```
1  725364   NA T  A  NA NA NA NA 1.015971 10.75 NA NA NA NA downstream_gene_variant RP11-206L10.9  ABCDEF   ABCDEF100   1  uncertain   PASS;ABHet=0.892;ABHom=0.983;AC=34;AF=3.688e-03;AN=9220;BaseQRankSum=0.553;DP=208471;FS=11.542;GC=38.10;GQ_MEAN=12.32;GQ_STDDEV=26.06;InbreedingCoeff=-0.2362;MLEAC=20;MLEAF=2.169e-03;MQ=57.56;MQ0=0;MQRankSum=-1.733e+00;NCC=0;NDA=5;OND=1.400e-03;QD=11.90;ReadPosRankSum=-1.134e+00;SOR=2.972;VQSLOD=-1.189e+01;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=RP11-206L10.9,RP11-206L10.8;GeneDetail=2312,dist=10895;genomicSuperDups=0.984251,chr1:224139760;CADD=1.015971;CADD_Phred=10.75;FATHMM_c=0.04301;FATHMM_nc=0.00108;CSQ=A|downstream_gene_variant|MODIFIER|RP11-206L10.9|ENSG00000237491|Transcript|ENST00000591702|lincRNA|||||||||||2312|1|SNV|Clone_based_vega_gene||YES|||||||||||||||||||||||||||||||||||||||||  0/0   .:52,0:52:0:.:.:0,0,1113   GT:AB:AD:DP:GQ:PGT:PID:PL  0  0.003968254 NA 0  NA 0  NA 0  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA 0.004816956 0  0  NA
1  725364   NA T  A  NA NA NA NA 1.015971 10.75 NA NA NA NA downstream_gene_variant RP11-206L10.9  ABCDEF   ABCDEF101   1  uncertain   PASS;ABHet=0.892;ABHom=0.983;AC=34;AF=3.688e-03;AN=9220;BaseQRankSum=0.553;DP=208471;FS=11.542;GC=38.10;GQ_MEAN=12.32;GQ_STDDEV=26.06;InbreedingCoeff=-0.2362;MLEAC=20;MLEAF=2.169e-03;MQ=57.56;MQ0=0;MQRankSum=-1.733e+00;NCC=0;NDA=5;OND=1.400e-03;QD=11.90;ReadPosRankSum=-1.134e+00;SOR=2.972;VQSLOD=-1.189e+01;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=RP11-206L10.9,RP11-206L10.8;GeneDetail=2312,dist=10895;genomicSuperDups=0.984251,chr1:224139760;CADD=1.015971;CADD_Phred=10.75;FATHMM_c=0.04301;FATHMM_nc=0.00108;CSQ=A|downstream_gene_variant|MODIFIER|RP11-206L10.9|ENSG00000237491|Transcript|ENST00000591702|lincRNA|||||||||||2312|1|SNV|Clone_based_vega_gene||YES|||||||||||||||||||||||||||||||||||||||||  0/0   .:26,0:26:0:.:.:0,0,421 GT:AB:AD:DP:GQ:PGT:PID:PL  0  0.003968254 NA 0  NA 0  NA 0  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA 0.004816956 0  0  NA
```
Chromosome 2 Input File (Shard 1): TS_filtering_examples/multiple_file_chr_shards/subdir_input/2/Transmission_Summary_Flat_chr2_00000_annotated_for_analysis.db
```
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
2  10205 NA T  G  NA NA NA NA NA NA NA NA NA NA regulatory_region_variant  NA ABCEFG   ABCEFG100   0  uncertain   PASS;ABHet=0.500;ABHom=NaN;AC=5;AF=1.132e-03;AN=4416;BaseQRankSum=0.804;DP=12674;FS=0.000;GC=61.90;GQ_MEAN=4.07;GQ_STDDEV=4.97;InbreedingCoeff=-0.1347;MLEAC=3;MLEAF=6.793e-04;MQ=59.67;MQ0=0;MQRankSum=0.804;NCC=2402;NDA=2;NEGATIVE_TRAIN_SITE;QD=34.24;ReadPosRankSum=0.804;SOR=0.538;VQSLOD=-1.159e+01;VariantType=SNP;culprit=DP;cytoBand=2p25.3;Func=intergenic;Gene=NONE,FAM110C;GeneDetail=NONE,dist=28609;FATHMM_c=0.21254;FATHMM_nc=0.36333;CSQ=G|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000675422|TF_binding_site||||||||||rs111214705|||SNV||||||||||||||||||||||||||||||||||||||||||||  0/0   .:3,0:3:0:0,0,40  GT:AB:AD:DP:GQ:PL 0  0.001088139 NA 0.001088139 NA 51.37566138 NA 2  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA
2  10205 NA T  G  NA NA NA NA NA NA NA NA NA NA regulatory_region_variant  NA ABCEFG   ABCEFG101   1  missing  PASS;ABHet=0.500;ABHom=NaN;AC=5;AF=1.132e-03;AN=4416;BaseQRankSum=0.804;DP=12674;FS=0.000;GC=61.90;GQ_MEAN=4.07;GQ_STDDEV=4.97;InbreedingCoeff=-0.1347;MLEAC=3;MLEAF=6.793e-04;MQ=59.67;MQ0=0;MQRankSum=0.804;NCC=2402;NDA=2;NEGATIVE_TRAIN_SITE;QD=34.24;ReadPosRankSum=0.804;SOR=0.538;VQSLOD=-1.159e+01;VariantType=SNP;culprit=DP;cytoBand=2p25.3;Func=intergenic;Gene=NONE,FAM110C;GeneDetail=NONE,dist=28609;FATHMM_c=0.21254;FATHMM_nc=0.36333;CSQ=G|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000675422|TF_binding_site||||||||||rs111214705|||SNV||||||||||||||||||||||||||||||||||||||||||||  ./.   .:0,0:0  GT:AB:AD:DP:GQ:PL 0  0.001088139 NA 0.001088139 NA 51.37566138 NA 2  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA
```
Chromosome 2 Input File (Shard 2): TS_filtering_examples/multiple_file_chr_shards/subdir_input/2/Transmission_Summary_Flat_chr2_00001_annotated_for_analysis.db
```
2  399561   NA T  C  NA NA NA NA NA NA NA NA NA NA intron_variant&non_coding_transcript_variant AC105393.1  ABCDEF   ABCDEF100   1  uncertain   PASS;ABHet=0.720;ABHom=0.998;AC=14;AF=1.661e-03;AN=8428;BaseQRankSum=-8.100e-01;DP=109926;FS=0.000;GC=47.62;GQ_MEAN=25.43;GQ_STDDEV=27.35;InbreedingCoeff=-0.0960;MLEAC=12;MLEAF=1.424e-03;MQ=46.95;MQ0=0;MQRankSum=-7.900e-02;NCC=396;NDA=3;OND=3.422e-05;QD=6.41;ReadPosRankSum=-1.160e-01;SOR=0.748;VQSLOD=-9.640e+00;VariantType=SNP;culprit=MQ;cytoBand=2p25.3;Func=ncRNA_intronic;Gene=AC105393.1;FATHMM_c=0.04423;FATHMM_nc=0.00123;CSQ=C|intron_variant&non_coding_transcript_variant|MODIFIER|AC105393.1|ENSG00000236856|Transcript|ENST00000431911|lincRNA||1/1|ENST00000431911.1:n.186+10964T>C|||||||rs71337652||1|SNV|Clone_based_vega_gene||YES|||||||||||||||||||||||||||||||||||||||||   0/0   .:11,0:11:0:.:.:0,0,179 GT:AB:AD:DP:GQ:PGT:PID:PL  0  0.002605675 NA 0  NA 8.624338624 NA 0  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA 0.002089864 0  7.803468208 NA
2  399561   NA T  C  NA NA NA NA NA NA NA NA NA NA intron_variant&non_coding_transcript_variant AC105393.1  ABCDEF   ABCDEF101   0  uncertain   PASS;ABHet=0.720;ABHom=0.998;AC=14;AF=1.661e-03;AN=8428;BaseQRankSum=-8.100e-01;DP=109926;FS=0.000;GC=47.62;GQ_MEAN=25.43;GQ_STDDEV=27.35;InbreedingCoeff=-0.0960;MLEAC=12;MLEAF=1.424e-03;MQ=46.95;MQ0=0;MQRankSum=-7.900e-02;NCC=396;NDA=3;OND=3.422e-05;QD=6.41;ReadPosRankSum=-1.160e-01;SOR=0.748;VQSLOD=-9.640e+00;VariantType=SNP;culprit=MQ;cytoBand=2p25.3;Func=ncRNA_intronic;Gene=AC105393.1;FATHMM_c=0.04423;FATHMM_nc=0.00123;CSQ=C|intron_variant&non_coding_transcript_variant|MODIFIER|AC105393.1|ENSG00000236856|Transcript|ENST00000431911|lincRNA||1/1|ENST00000431911.1:n.186+10964T>C|||||||rs71337652||1|SNV|Clone_based_vega_gene||YES|||||||||||||||||||||||||||||||||||||||||   0/0   .:10,0:10:6:.:.:0,6,206 GT:AB:AD:DP:GQ:PGT:PID:PL  0  0.002605675 NA 0  NA 8.624338624 NA 0  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA 0.002089864 0  7.803468208 NA
```

Chromosome 3 Input File (Shard 1): TS_filtering_examples/multiple_file_chr_shards/subdir_input/3/Transmission_Summary_Flat_chr3_00000_annotated_for_analysis.db
```
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
3  60082 NA G  A  NA NA NA NA NA NA NA NA NA NA intergenic_variant   NA ABCDEF   ABCDEF100   1  uncertain   PASS;ABHet=0.556;AC=2;AF=2.169e-04;AN=9220;BaseQRankSum=-2.890e-01;DP=58911;FS=0.000;GC=38.10;GQ_MEAN=37.95;GQ_STDDEV=12.17;InbreedingCoeff=-0.0017;MLEAC=2;MLEAF=2.169e-04;MQ=50.28;MQ0=0;MQRankSum=1.84;NCC=0;NDA=1;QD=6.89;ReadPosRankSum=0.361;SOR=1.329;VQSLOD=-1.253e+01;VariantType=SNP;culprit=MQ;cytoBand=3p26.3;Func=intergenic;Gene=NONE,AY269186.1;GeneDetail=NONE,dist=5349;genomicSuperDups=0.95747,chr3:75279192;FATHMM_c=0.07045;FATHMM_nc=0.00765;CSQ=A|intergenic_variant|MODIFIER||||||||||||||||||SNV||||||||||||||||||||||||||||||||||||||||||||   0/0   .:7,0:7:21:0,21,224  GT:AB:AD:DP:GQ:PL 4.92E-05 0.00026455  NA 0  NA 0  NA 5  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA 4.92E-05
3  60082 NA G  A  NA NA NA NA NA NA NA NA NA NA intergenic_variant   NA ABCDEF   ABCDEF101   1  uncertain   PASS;ABHet=0.556;AC=2;AF=2.169e-04;AN=9220;BaseQRankSum=-2.890e-01;DP=58911;FS=0.000;GC=38.10;GQ_MEAN=37.95;GQ_STDDEV=12.17;InbreedingCoeff=-0.0017;MLEAC=2;MLEAF=2.169e-04;MQ=50.28;MQ0=0;MQRankSum=1.84;NCC=0;NDA=1;QD=6.89;ReadPosRankSum=0.361;SOR=1.329;VQSLOD=-1.253e+01;VariantType=SNP;culprit=MQ;cytoBand=3p26.3;Func=intergenic;Gene=NONE,AY269186.1;GeneDetail=NONE,dist=5349;genomicSuperDups=0.95747,chr3:75279192;FATHMM_c=0.07045;FATHMM_nc=0.00765;CSQ=A|intergenic_variant|MODIFIER||||||||||||||||||SNV||||||||||||||||||||||||||||||||||||||||||||   0/0   .:6,0:6:18:0,18,193  GT:AB:AD:DP:GQ:PL 4.92E-05 0.00026455  NA 0  NA 0  NA 5  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA 4.92E-05
```
Chromosome 3 Input File (Shard 2): TS_filtering_examples/multiple_file_chr_shards/subdir_input/3/Transmission_Summary_Flat_chr3_00001_annotated_for_analysis.db
```
3  375758   NA A  G  NA NA 0.871805 0.935 NA NA NA NA NA NA intron_variant CHL1  ABCDEF   ABCDEF100   1  uncertain   PASS;ABHet=0.498;ABHom=1.000;AC=7953;AF=0.863;AN=9220;BaseQRankSum=0.950;DP=169494;FS=0.000;GC=47.62;GQ_MEAN=171.22;GQ_STDDEV=129.85;InbreedingCoeff=0.0383;MLEAC=7953;MLEAF=0.863;MQ=60.00;MQ0=0;MQRankSum=0.091;NCC=0;NDA=2;OND=3.687e-04;POSITIVE_TRAIN_SITE;QD=27.55;ReadPosRankSum=0.446;SOR=0.728;VQSLOD=23.30;VariantType=SNP;culprit=MQ;cytoBand=3p26.3;Func=intronic;Gene=CHL1;FATHMM_c=0.02636;FATHMM_nc=0.00031;GWAVA_region=0.23;GWAVA_tss=0.07;GWAVA_unmatched=182;cg46=0.935;1000g2015aug_all=0.871805;CSQ=G|intron_variant|MODIFIER|CHL1|ENSG00000134121|Transcript|ENST00000256509|protein_coding||5/27|ENST00000256509.2:c.385+5721A>G|||||||rs9858162||1|SNV|HGNC|1939|YES|||CCDS2556.1|ENSP00000256509|CHL1_HUMAN|C9JW79_HUMAN&C9JH37_HUMAN&C9JEY3_HUMAN&C9J905_HUMAN|UPI000013CF0F|1|||||A:0.1282|G:0.8192|G:0.9049|G:0.9911|G:0.836|G:0.8333|||||||||||||||||||||||  1-Jan .:0,29:29:86:.:.:891,86,0  GT:AB:AD:DP:GQ:PGT:PID:PL  0.935 0.861111111 0.709762533 0.746031746 0  0  NA 0  CHL1  NA 41.92 9.51  23.3  44.85 -1.853586132   0.0003275   0.999647383 2.51E-05 0.133650262 chr3:238279-451090   NA NA NA 0.861271676 0.738921002 0  NA
3  375758   NA A  G  NA NA 0.871805 0.935 NA NA NA NA NA NA intron_variant CHL1  ABCDEF   ABCDEF101   1  uncertain   PASS;ABHet=0.498;ABHom=1.000;AC=7953;AF=0.863;AN=9220;BaseQRankSum=0.950;DP=169494;FS=0.000;GC=47.62;GQ_MEAN=171.22;GQ_STDDEV=129.85;InbreedingCoeff=0.0383;MLEAC=7953;MLEAF=0.863;MQ=60.00;MQ0=0;MQRankSum=0.091;NCC=0;NDA=2;OND=3.687e-04;POSITIVE_TRAIN_SITE;QD=27.55;ReadPosRankSum=0.446;SOR=0.728;VQSLOD=23.30;VariantType=SNP;culprit=MQ;cytoBand=3p26.3;Func=intronic;Gene=CHL1;FATHMM_c=0.02636;FATHMM_nc=0.00031;GWAVA_region=0.23;GWAVA_tss=0.07;GWAVA_unmatched=182;cg46=0.935;1000g2015aug_all=0.871805;CSQ=G|intron_variant|MODIFIER|CHL1|ENSG00000134121|Transcript|ENST00000256509|protein_coding||5/27|ENST00000256509.2:c.385+5721A>G|||||||rs9858162||1|SNV|HGNC|1939|YES|||CCDS2556.1|ENSP00000256509|CHL1_HUMAN|C9JW79_HUMAN&C9JH37_HUMAN&C9JEY3_HUMAN&C9J905_HUMAN|UPI000013CF0F|1|||||A:0.1282|G:0.8192|G:0.9049|G:0.9911|G:0.836|G:0.8333|||||||||||||||||||||||  1-Jan .:0,24:24:72:.:.:783,72,0  GT:AB:AD:DP:GQ:PGT:PID:PL  9.35E-01 0.861111111 0.709762533 0.746031746 0  0  NA 0  CHL1  NA 41.92 9.51  23.3  44.85 -1.853586132   0.0003275   0.999647383 2.51E-05 0.133650262 chr3:238279-451090   NA NA NA 0.861271676 0.738921002 0  NA
```

**General Command**:
```
bash Filtering_Wrapper_Script.sh --input_dir <input_dir> --input_prefix <input_prefix> --input_suffix <input_prefix> --output_dir <output_dir> --input_type chr_shards --flat_or_family <flat_or_family (e.g. flat, family)> [--comb_and_clean <comb_and_clean (e.g. merge_all_chr, merge_by_chr, or no_comb_clean)>] [--no_qsub <no_qsub (e.g. True, False)>] [--log_dir <log_dir>] [--exclude_chr <exclude_chr>]
```

**Merge All Chromosome Command**:
```
bash /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/Filtering_Wrapper_Script.sh --input_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/multiple_file_chr_shards/subdir_input --input_prefix Transmission_Summary_Flat_chr --input_suffix _annotated_for_analysis.db --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/merge_all_chr_output --input_type chr_shards --flat_or_family flat --comb_and_clean merge_all_chr --consequence upstream_gene_variant
```

**Merge All Chromosome Output File**: TS_filtering_examples/multiple_file_chr_shards/merge_all_chr_output/Transmission_Summary_Flat_chr_MergedAllChr_annotated_for_analysis_filtered.db
```
#bmap command: /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/filter_flat_annotated_transmission_summary.py --input_db /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/subdir_input/1/Transmission_Summary_Flat_chr1_00000_annotated_for_analysis.db --input_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/subdir_input --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/merge_all_chr_output/1 --comb_and_clean no_comb_clean --input_prefix Transmission_Summary_Flat_chr --input_suffix _annotated_for_analysis.db --input_type chr_shards --flat_or_family flat --consequence upstream_gene_variant
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
1  10492 NA C  T  NA NA NA NA 0.99208  10.62 NA NA NA NA upstream_gene_variant   DDX11L1  ABCDEF   ABCDEF100   1  uncertain   PASS;ABHet=0.734;ABHom=0.953;AC=1095;AF=0.119;AN=9170;BaseQRankSum=-9.500e-01;DP=151995;FS=1.412;GC=95.24;GQ_MEAN=56.05;GQ_STDDEV=42.80;InbreedingCoeff=-0.1477;MLEAC=1158;MLEAF=0.126;MQ=53.11;MQ0=0;MQRankSum=0.043;NCC=25;NDA=1;OND=5.382e-03;QD=3.83;ReadPosRankSum=0.00;SOR=0.445;VQSLOD=-5.762e+00;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=NONE,DDX11L1;GeneDetail=NONE,dist=1377;genomicSuperDups=0.993729,chr9:10843;CADD=0.992080;CADD_Phred=10.62;FATHMM_c=0.73423;FATHMM_nc=0.05648;GWAVA_region=0.4;GWAVA_tss=0.26;GWAVA_unmatched=182;CSQ=T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000456328|processed_transcript||||||||||rs55998931|1377|1|SNV|HGNC|37102|YES|||||||||||||||||||||||||||||||||||||||||  0/1   0.330:3,6:9:67:.:.:162,0,67   GT:AB:AD:DP:GQ:PGT:PID:PL  0.0811632   0.120671284 0  0.004794885 0.527704485 0.687830688 NA 1  DDX11L1  ambiguous_previous   NA NA NA NA NA NA NA NA NA NA NA NA NA 0.147540984 0.00192864  0.096339114 8.12E-02
1  10492 NA C  T  NA NA NA NA 0.99208  10.62 NA NA NA NA upstream_gene_variant   DDX11L1  ABCDEF   ABCDEF101   1  uncertain   PASS;ABHet=0.734;ABHom=0.953;AC=1095;AF=0.119;AN=9170;BaseQRankSum=-9.500e-01;DP=151995;FS=1.412;GC=95.24;GQ_MEAN=56.05;GQ_STDDEV=42.80;InbreedingCoeff=-0.1477;MLEAC=1158;MLEAF=0.126;MQ=53.11;MQ0=0;MQRankSum=0.043;NCC=25;NDA=1;OND=5.382e-03;QD=3.83;ReadPosRankSum=0.00;SOR=0.445;VQSLOD=-5.762e+00;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=NONE,DDX11L1;GeneDetail=NONE,dist=1377;genomicSuperDups=0.993729,chr9:10843;CADD=0.992080;CADD_Phred=10.62;FATHMM_c=0.73423;FATHMM_nc=0.05648;GWAVA_region=0.4;GWAVA_tss=0.26;GWAVA_unmatched=182;CSQ=T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000456328|processed_transcript||||||||||rs55998931|1377|1|SNV|HGNC|37102|YES|||||||||||||||||||||||||||||||||||||||||  0/0   .:23,0:23:0:.:.:0,0,36  GT:AB:AD:DP:GQ:PGT:PID:PL  0.0811632   0.120671284 0  0.004794885 0.527704485 0.687830688 NA 1  DDX11L1  ambiguous_previous   NA NA NA NA NA NA NA NA NA NA NA NA NA 0.147540984 0.00192864  0.096339114 8.12E-02
```

**Merge By Chromosome Command**:
```
bash /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/Filtering_Wrapper_Script.sh --input_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/subdir_input --input_prefix Transmission_Summary_Flat_chr --input_suffix _annotated_for_analysis.db --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/merge_all_chr_output --input_type chr_shards --flat_or_family flat --comb_and_clean merge_by_chr --consequence upstream_gene_variant
```

**Merge By Chromosome Output Files**:

Chromosome 1 Output File: TS_filtering_examples/multiple_file_chr_shards/merge_by_chr_output/Transmission_Summary_Flat_chr1_annotated_for_analysis_filtered.db
```
#bmap command: /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/filter_flat_annotated_transmission_summary.py --input_db /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/subdir_input/1/Transmission_Summary_Flat_chr1_00000_annotated_for_analysis.db --input_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/subdir_input --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/merge_by_chr_output/1 --comb_and_clean no_comb_clean --input_prefix Transmission_Summary_Flat_chr --input_suffix _annotated_for_analysis.db --input_type chr_shards --flat_or_family flat --consequence upstream_gene_variant
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
1  10492 NA C  T  NA NA NA NA 0.99208  10.62 NA NA NA NA upstream_gene_variant   DDX11L1  ABCDEF   ABCDEF100   1  uncertain   PASS;ABHet=0.734;ABHom=0.953;AC=1095;AF=0.119;AN=9170;BaseQRankSum=-9.500e-01;DP=151995;FS=1.412;GC=95.24;GQ_MEAN=56.05;GQ_STDDEV=42.80;InbreedingCoeff=-0.1477;MLEAC=1158;MLEAF=0.126;MQ=53.11;MQ0=0;MQRankSum=0.043;NCC=25;NDA=1;OND=5.382e-03;QD=3.83;ReadPosRankSum=0.00;SOR=0.445;VQSLOD=-5.762e+00;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=NONE,DDX11L1;GeneDetail=NONE,dist=1377;genomicSuperDups=0.993729,chr9:10843;CADD=0.992080;CADD_Phred=10.62;FATHMM_c=0.73423;FATHMM_nc=0.05648;GWAVA_region=0.4;GWAVA_tss=0.26;GWAVA_unmatched=182;CSQ=T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000456328|processed_transcript||||||||||rs55998931|1377|1|SNV|HGNC|37102|YES|||||||||||||||||||||||||||||||||||||||||  0/1   0.330:3,6:9:67:.:.:162,0,67   GT:AB:AD:DP:GQ:PGT:PID:PL  0.0811632   0.120671284 0  0.004794885 0.527704485 0.687830688 NA 1  DDX11L1  ambiguous_previous   NA NA NA NA NA NA NA NA NA NA NA NA NA 0.147540984 0.00192864  0.096339114 8.12E-02
1  10492 NA C  T  NA NA NA NA 0.99208  10.62 NA NA NA NA upstream_gene_variant   DDX11L1  ABCDEF   ABCDEF101   1  uncertain   PASS;ABHet=0.734;ABHom=0.953;AC=1095;AF=0.119;AN=9170;BaseQRankSum=-9.500e-01;DP=151995;FS=1.412;GC=95.24;GQ_MEAN=56.05;GQ_STDDEV=42.80;InbreedingCoeff=-0.1477;MLEAC=1158;MLEAF=0.126;MQ=53.11;MQ0=0;MQRankSum=0.043;NCC=25;NDA=1;OND=5.382e-03;QD=3.83;ReadPosRankSum=0.00;SOR=0.445;VQSLOD=-5.762e+00;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=NONE,DDX11L1;GeneDetail=NONE,dist=1377;genomicSuperDups=0.993729,chr9:10843;CADD=0.992080;CADD_Phred=10.62;FATHMM_c=0.73423;FATHMM_nc=0.05648;GWAVA_region=0.4;GWAVA_tss=0.26;GWAVA_unmatched=182;CSQ=T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000456328|processed_transcript||||||||||rs55998931|1377|1|SNV|HGNC|37102|YES|||||||||||||||||||||||||||||||||||||||||  0/0   .:23,0:23:0:.:.:0,0,36  GT:AB:AD:DP:GQ:PGT:PID:PL  0.0811632   0.120671284 0  0.004794885 0.527704485 0.687830688 NA 1  DDX11L1  ambiguous_previous   NA NA NA NA NA NA NA NA NA NA NA NA NA 0.147540984 0.00192864  0.096339114 8.12E-02
```

Chromosome 2 Output File: TS_filtering_examples/multiple_file_chr_shards/merge_by_chr_output/Transmission_Summary_Flat_chr2_annotated_for_analysis_filtered.db
```
#bmap command: /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/filter_flat_annotated_transmission_summary.py --input_db /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/subdir_input/2/Transmission_Summary_Flat_chr2_00000_annotated_for_analysis.db --input_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/multiple_file_chr_shards/subdir_input --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/merge_by_chr_output/2 --comb_and_clean no_comb_clean --input_prefix Transmission_Summary_Flat_chr --input_suffix _annotated_for_analysis.db --input_type chr_shards --flat_or_family flat --consequence upstream_gene_variant
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
   - Header only provided, since no variants met filtering requirements.
```

Chromosome 3 Output File: TS_filtering_examples/multiple_file_chr_shards/merge_by_chr_output/Transmission_Summary_Flat_chr3_annotated_for_analysis_filtered.db
```
#bmap command: /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/filter_flat_annotated_transmission_summary.py --input_db /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/subdir_input/3/Transmission_Summary_Flat_chr3_00000_annotated_for_analysis.db --input_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/subdir_input --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/merge_by_chr_output/3 --comb_and_clean no_comb_clean --input_prefix Transmission_Summary_Flat_chr --input_suffix _annotated_for_analysis.db --input_type chr_shards --flat_or_family flat --consequence upstream_gene_variant
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
   - Header only provided, since no variants met filtering requirements.
```

**No Combine and Clean Chromosome Command**:
```
bash /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/Filtering_Wrapper_Script.sh --input_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/subdir_input --input_prefix Transmission_Summary_Flat_chr --input_suffix _annotated_for_analysis.db --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/no_comb_clean_output --input_type chr_shards --flat_or_family flat --comb_and_clean no_comb_clean --consequence upstream_gene_variant
```

**No Combine and Clean Chromosome Output Files**:

Chromosome 1 Output File (shard 1): TS_filtering_examples/multiple_file_chr_shards/no_comb_clean_output/1/Transmission_Summary_Flat_chr1_00000_annotated_for_analysis_filtered.db
```
Empty, since no variants met filtering requirments.
```

Chromosome 1 Output File (shard 2): TS_filtering_examples/multiple_file_chr_shards/no_comb_clean_output/1/Transmission_Summary_Flat_chr1_00001_annotated_for_analysis_filtered.db
```
#bmap command: /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/filter_flat_annotated_transmission_summary.py --input_db /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/subdir_input/1/Transmission_Summary_Flat_chr1_00000_annotated_for_analysis.db --input_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/subdir_input --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/no_comb_clean_output/1 --comb_and_clean no_comb_clean --input_prefix Transmission_Summary_Flat_chr --input_suffix _annotated_for_analysis.db --input_type chr_shards --flat_or_family flat --consequence upstream_gene_variant
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
1  10492 NA C  T  NA NA NA NA 0.99208  10.62 NA NA NA NA upstream_gene_variant   DDX11L1  ABCDEF   ABCDEF100   1  uncertain   PASS;ABHet=0.734;ABHom=0.953;AC=1095;AF=0.119;AN=9170;BaseQRankSum=-9.500e-01;DP=151995;FS=1.412;GC=95.24;GQ_MEAN=56.05;GQ_STDDEV=42.80;InbreedingCoeff=-0.1477;MLEAC=1158;MLEAF=0.126;MQ=53.11;MQ0=0;MQRankSum=0.043;NCC=25;NDA=1;OND=5.382e-03;QD=3.83;ReadPosRankSum=0.00;SOR=0.445;VQSLOD=-5.762e+00;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=NONE,DDX11L1;GeneDetail=NONE,dist=1377;genomicSuperDups=0.993729,chr9:10843;CADD=0.992080;CADD_Phred=10.62;FATHMM_c=0.73423;FATHMM_nc=0.05648;GWAVA_region=0.4;GWAVA_tss=0.26;GWAVA_unmatched=182;CSQ=T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000456328|processed_transcript||||||||||rs55998931|1377|1|SNV|HGNC|37102|YES|||||||||||||||||||||||||||||||||||||||||  0/1   0.330:3,6:9:67:.:.:162,0,67   GT:AB:AD:DP:GQ:PGT:PID:PL  0.0811632   0.120671284 0  0.004794885 0.527704485 0.687830688 NA 1  DDX11L1  ambiguous_previous   NA NA NA NA NA NA NA NA NA NA NA NA NA 0.147540984 0.00192864  0.096339114 8.12E-02
1  10492 NA C  T  NA NA NA NA 0.99208  10.62 NA NA NA NA upstream_gene_variant   DDX11L1  ABCDEF   ABCDEF101   1  uncertain   PASS;ABHet=0.734;ABHom=0.953;AC=1095;AF=0.119;AN=9170;BaseQRankSum=-9.500e-01;DP=151995;FS=1.412;GC=95.24;GQ_MEAN=56.05;GQ_STDDEV=42.80;InbreedingCoeff=-0.1477;MLEAC=1158;MLEAF=0.126;MQ=53.11;MQ0=0;MQRankSum=0.043;NCC=25;NDA=1;OND=5.382e-03;QD=3.83;ReadPosRankSum=0.00;SOR=0.445;VQSLOD=-5.762e+00;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=NONE,DDX11L1;GeneDetail=NONE,dist=1377;genomicSuperDups=0.993729,chr9:10843;CADD=0.992080;CADD_Phred=10.62;FATHMM_c=0.73423;FATHMM_nc=0.05648;GWAVA_region=0.4;GWAVA_tss=0.26;GWAVA_unmatched=182;CSQ=T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000456328|processed_transcript||||||||||rs55998931|1377|1|SNV|HGNC|37102|YES|||||||||||||||||||||||||||||||||||||||||  0/0   .:23,0:23:0:.:.:0,0,36  GT:AB:AD:DP:GQ:PGT:PID:PL  0.0811632   0.120671284 0  0.004794885 0.527704485 0.687830688 NA 1  DDX11L1  ambiguous_previous   NA NA NA NA NA NA NA NA NA NA NA NA NA 0.147540984 0.00192864  0.096339114 8.12E-02

```

Chromosome 2 Output File (shard 1): TS_filtering_examples/multiple_file_chr_shards/no_comb_clean_output/2/Transmission_Summary_Flat_chr2_00000_annotated_for_analysis_filtered.db
```
#bmap command: /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/filter_flat_annotated_transmission_summary.py --input_db /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/multiple_file_chr_shards/subdir_input/2/Transmission_Summary_Flat_chr2_00000_annotated_for_analysis.db --input_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/multiple_file_chr_shards/subdir_input --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/multiple_file_chr_shards/no_comb_clean_output/2 --comb_and_clean no_comb_clean --input_prefix Transmission_Summary_Flat_chr --input_suffix _annotated_for_analysis.db --input_type chr_shards --flat_or_family flat --consequence upstream_gene_variant
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
   - Header only provided, since no variants met filtering requirements.
```

Chromosome 2 Output File (shard 2): TS_filtering_examples/multiple_file_chr_shards/no_comb_clean_output/2/Transmission_Summary_Flat_chr2_00001_annotated_for_analysis_filtered.db
```
Empty, since no variants met filtering requirments.
```

Chromosome 3 Output File (shard 1): TS_filtering_examples/multiple_file_chr_shards/no_comb_clean_output/3/Transmission_Summary_Flat_chr3_00000_annotated_for_analysis_filtered.db
```
#bmap command: /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/filter_flat_annotated_transmission_summary.py --input_db /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/subdir_input/3/Transmission_Summary_Flat_chr3_00000_annotated_for_analysis.db --input_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/subdir_input --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/no_comb_clean_output/3 --comb_and_clean no_comb_clean --input_prefix Transmission_Summary_Flat_chr --input_suffix _annotated_for_analysis.db --input_type chr_shards --flat_or_family flat --consequence upstream_gene_variant
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
   - Header only provided, since no variants met filtering requirements.
```
Chromosome 3 Output File (shard 2): TS_filtering_examples/multiple_file_chr_shards/no_comb_clean_output/3/Transmission_Summary_Flat_chr3_00001_annotated_for_analysis_filtered.db
```
Empty, since no variants met filtering requirments.
```

**Input Files Within input_dir**: <br/>

* Note: Input files can be in within input_dir 

Chromosome 1 Input File (Shard 1): TS_filtering_examples/multiple_file_chr_shards/main_dir_input/Transmission_Summary_Flat_chr1_00000_annotated_for_analysis.db 
```
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
1  10492 NA C  T  NA NA NA NA 0.99208  10.62 NA NA NA NA upstream_gene_variant   DDX11L1  ABCDEF   ABCDEF100   1  uncertain   PASS;ABHet=0.734;ABHom=0.953;AC=1095;AF=0.119;AN=9170;BaseQRankSum=-9.500e-01;DP=151995;FS=1.412;GC=95.24;GQ_MEAN=56.05;GQ_STDDEV=42.80;InbreedingCoeff=-0.1477;MLEAC=1158;MLEAF=0.126;MQ=53.11;MQ0=0;MQRankSum=0.043;NCC=25;NDA=1;OND=5.382e-03;QD=3.83;ReadPosRankSum=0.00;SOR=0.445;VQSLOD=-5.762e+00;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=NONE,DDX11L1;GeneDetail=NONE,dist=1377;genomicSuperDups=0.993729,chr9:10843;CADD=0.992080;CADD_Phred=10.62;FATHMM_c=0.73423;FATHMM_nc=0.05648;GWAVA_region=0.4;GWAVA_tss=0.26;GWAVA_unmatched=182;CSQ=T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000456328|processed_transcript||||||||||rs55998931|1377|1|SNV|HGNC|37102|YES|||||||||||||||||||||||||||||||||||||||||  0/1   0.330:3,6:9:67:.:.:162,0,67   GT:AB:AD:DP:GQ:PGT:PID:PL  0.0811632   0.120671284 0  0.004794885 0.527704485 0.687830688 NA 1  DDX11L1  ambiguous_previous   NA NA NA NA NA NA NA NA NA NA NA NA NA 0.147540984 0.00192864  0.096339114 8.12E-02
1  10492 NA C  T  NA NA NA NA 0.99208  10.62 NA NA NA NA upstream_gene_variant   DDX11L1  ABCDEF   ABCDEF101   1  uncertain   PASS;ABHet=0.734;ABHom=0.953;AC=1095;AF=0.119;AN=9170;BaseQRankSum=-9.500e-01;DP=151995;FS=1.412;GC=95.24;GQ_MEAN=56.05;GQ_STDDEV=42.80;InbreedingCoeff=-0.1477;MLEAC=1158;MLEAF=0.126;MQ=53.11;MQ0=0;MQRankSum=0.043;NCC=25;NDA=1;OND=5.382e-03;QD=3.83;ReadPosRankSum=0.00;SOR=0.445;VQSLOD=-5.762e+00;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=NONE,DDX11L1;GeneDetail=NONE,dist=1377;genomicSuperDups=0.993729,chr9:10843;CADD=0.992080;CADD_Phred=10.62;FATHMM_c=0.73423;FATHMM_nc=0.05648;GWAVA_region=0.4;GWAVA_tss=0.26;GWAVA_unmatched=182;CSQ=T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000456328|processed_transcript||||||||||rs55998931|1377|1|SNV|HGNC|37102|YES|||||||||||||||||||||||||||||||||||||||||  0/0   .:23,0:23:0:.:.:0,0,36  GT:AB:AD:DP:GQ:PGT:PID:PL  0.0811632   0.120671284 0  0.004794885 0.527704485 0.687830688 NA 1  DDX11L1  ambiguous_previous   NA NA NA NA NA NA NA NA NA NA NA NA NA 0.147540984 0.00192864  0.096339114 8.12E-02
```
Chromosome 1 Input File (Shard 2): TS_filtering_examples/multiple_file_chr_shards/main_dir_input/Transmission_Summary_Flat_chr1_00001_annotated_for_analysis.db 
```
1  725364   NA T  A  NA NA NA NA 1.015971 10.75 NA NA NA NA downstream_gene_variant RP11-206L10.9  ABCDEF   ABCDEF100   1  uncertain   PASS;ABHet=0.892;ABHom=0.983;AC=34;AF=3.688e-03;AN=9220;BaseQRankSum=0.553;DP=208471;FS=11.542;GC=38.10;GQ_MEAN=12.32;GQ_STDDEV=26.06;InbreedingCoeff=-0.2362;MLEAC=20;MLEAF=2.169e-03;MQ=57.56;MQ0=0;MQRankSum=-1.733e+00;NCC=0;NDA=5;OND=1.400e-03;QD=11.90;ReadPosRankSum=-1.134e+00;SOR=2.972;VQSLOD=-1.189e+01;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=RP11-206L10.9,RP11-206L10.8;GeneDetail=2312,dist=10895;genomicSuperDups=0.984251,chr1:224139760;CADD=1.015971;CADD_Phred=10.75;FATHMM_c=0.04301;FATHMM_nc=0.00108;CSQ=A|downstream_gene_variant|MODIFIER|RP11-206L10.9|ENSG00000237491|Transcript|ENST00000591702|lincRNA|||||||||||2312|1|SNV|Clone_based_vega_gene||YES|||||||||||||||||||||||||||||||||||||||||  0/0   .:52,0:52:0:.:.:0,0,1113   GT:AB:AD:DP:GQ:PGT:PID:PL  0  0.003968254 NA 0  NA 0  NA 0  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA 0.004816956 0  0  NA
1  725364   NA T  A  NA NA NA NA 1.015971 10.75 NA NA NA NA downstream_gene_variant RP11-206L10.9  ABCDEF   ABCDEF101   1  uncertain   PASS;ABHet=0.892;ABHom=0.983;AC=34;AF=3.688e-03;AN=9220;BaseQRankSum=0.553;DP=208471;FS=11.542;GC=38.10;GQ_MEAN=12.32;GQ_STDDEV=26.06;InbreedingCoeff=-0.2362;MLEAC=20;MLEAF=2.169e-03;MQ=57.56;MQ0=0;MQRankSum=-1.733e+00;NCC=0;NDA=5;OND=1.400e-03;QD=11.90;ReadPosRankSum=-1.134e+00;SOR=2.972;VQSLOD=-1.189e+01;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=RP11-206L10.9,RP11-206L10.8;GeneDetail=2312,dist=10895;genomicSuperDups=0.984251,chr1:224139760;CADD=1.015971;CADD_Phred=10.75;FATHMM_c=0.04301;FATHMM_nc=0.00108;CSQ=A|downstream_gene_variant|MODIFIER|RP11-206L10.9|ENSG00000237491|Transcript|ENST00000591702|lincRNA|||||||||||2312|1|SNV|Clone_based_vega_gene||YES|||||||||||||||||||||||||||||||||||||||||  0/0   .:26,0:26:0:.:.:0,0,421 GT:AB:AD:DP:GQ:PGT:PID:PL  0  0.003968254 NA 0  NA 0  NA 0  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA 0.004816956 0  0  NA
```
Chromosome 2 Input File (Shard 1): TS_filtering_examples/multiple_file_chr_shards/main_dir_input/Transmission_Summary_Flat_chr2_00000_annotated_for_analysis.db 
```
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
2  10205 NA T  G  NA NA NA NA NA NA NA NA NA NA regulatory_region_variant  NA ABCEFG   ABCEFG100   0  uncertain   PASS;ABHet=0.500;ABHom=NaN;AC=5;AF=1.132e-03;AN=4416;BaseQRankSum=0.804;DP=12674;FS=0.000;GC=61.90;GQ_MEAN=4.07;GQ_STDDEV=4.97;InbreedingCoeff=-0.1347;MLEAC=3;MLEAF=6.793e-04;MQ=59.67;MQ0=0;MQRankSum=0.804;NCC=2402;NDA=2;NEGATIVE_TRAIN_SITE;QD=34.24;ReadPosRankSum=0.804;SOR=0.538;VQSLOD=-1.159e+01;VariantType=SNP;culprit=DP;cytoBand=2p25.3;Func=intergenic;Gene=NONE,FAM110C;GeneDetail=NONE,dist=28609;FATHMM_c=0.21254;FATHMM_nc=0.36333;CSQ=G|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000675422|TF_binding_site||||||||||rs111214705|||SNV||||||||||||||||||||||||||||||||||||||||||||  0/0   .:3,0:3:0:0,0,40  GT:AB:AD:DP:GQ:PL 0  0.001088139 NA 0.001088139 NA 51.37566138 NA 2  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA
2  10205 NA T  G  NA NA NA NA NA NA NA NA NA NA regulatory_region_variant  NA ABCEFG   ABCEFG101   1  missing  PASS;ABHet=0.500;ABHom=NaN;AC=5;AF=1.132e-03;AN=4416;BaseQRankSum=0.804;DP=12674;FS=0.000;GC=61.90;GQ_MEAN=4.07;GQ_STDDEV=4.97;InbreedingCoeff=-0.1347;MLEAC=3;MLEAF=6.793e-04;MQ=59.67;MQ0=0;MQRankSum=0.804;NCC=2402;NDA=2;NEGATIVE_TRAIN_SITE;QD=34.24;ReadPosRankSum=0.804;SOR=0.538;VQSLOD=-1.159e+01;VariantType=SNP;culprit=DP;cytoBand=2p25.3;Func=intergenic;Gene=NONE,FAM110C;GeneDetail=NONE,dist=28609;FATHMM_c=0.21254;FATHMM_nc=0.36333;CSQ=G|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000675422|TF_binding_site||||||||||rs111214705|||SNV||||||||||||||||||||||||||||||||||||||||||||  ./.   .:0,0:0  GT:AB:AD:DP:GQ:PL 0  0.001088139 NA 0.001088139 NA 51.37566138 NA 2  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA
```
Chromosome 2 Input File (Shard 2): TS_filtering_examples/multiple_file_chr_shards/main_dir_input/Transmission_Summary_Flat_chr2_00001_annotated_for_analysis.db 
```
2  399561   NA T  C  NA NA NA NA NA NA NA NA NA NA intron_variant&non_coding_transcript_variant AC105393.1  ABCDEF   ABCDEF100   1  uncertain   PASS;ABHet=0.720;ABHom=0.998;AC=14;AF=1.661e-03;AN=8428;BaseQRankSum=-8.100e-01;DP=109926;FS=0.000;GC=47.62;GQ_MEAN=25.43;GQ_STDDEV=27.35;InbreedingCoeff=-0.0960;MLEAC=12;MLEAF=1.424e-03;MQ=46.95;MQ0=0;MQRankSum=-7.900e-02;NCC=396;NDA=3;OND=3.422e-05;QD=6.41;ReadPosRankSum=-1.160e-01;SOR=0.748;VQSLOD=-9.640e+00;VariantType=SNP;culprit=MQ;cytoBand=2p25.3;Func=ncRNA_intronic;Gene=AC105393.1;FATHMM_c=0.04423;FATHMM_nc=0.00123;CSQ=C|intron_variant&non_coding_transcript_variant|MODIFIER|AC105393.1|ENSG00000236856|Transcript|ENST00000431911|lincRNA||1/1|ENST00000431911.1:n.186+10964T>C|||||||rs71337652||1|SNV|Clone_based_vega_gene||YES|||||||||||||||||||||||||||||||||||||||||   0/0   .:11,0:11:0:.:.:0,0,179 GT:AB:AD:DP:GQ:PGT:PID:PL  0  0.002605675 NA 0  NA 8.624338624 NA 0  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA 0.002089864 0  7.803468208 NA
2  399561   NA T  C  NA NA NA NA NA NA NA NA NA NA intron_variant&non_coding_transcript_variant AC105393.1  ABCDEF   ABCDEF101   0  uncertain   PASS;ABHet=0.720;ABHom=0.998;AC=14;AF=1.661e-03;AN=8428;BaseQRankSum=-8.100e-01;DP=109926;FS=0.000;GC=47.62;GQ_MEAN=25.43;GQ_STDDEV=27.35;InbreedingCoeff=-0.0960;MLEAC=12;MLEAF=1.424e-03;MQ=46.95;MQ0=0;MQRankSum=-7.900e-02;NCC=396;NDA=3;OND=3.422e-05;QD=6.41;ReadPosRankSum=-1.160e-01;SOR=0.748;VQSLOD=-9.640e+00;VariantType=SNP;culprit=MQ;cytoBand=2p25.3;Func=ncRNA_intronic;Gene=AC105393.1;FATHMM_c=0.04423;FATHMM_nc=0.00123;CSQ=C|intron_variant&non_coding_transcript_variant|MODIFIER|AC105393.1|ENSG00000236856|Transcript|ENST00000431911|lincRNA||1/1|ENST00000431911.1:n.186+10964T>C|||||||rs71337652||1|SNV|Clone_based_vega_gene||YES|||||||||||||||||||||||||||||||||||||||||   0/0   .:10,0:10:6:.:.:0,6,206 GT:AB:AD:DP:GQ:PGT:PID:PL  0  0.002605675 NA 0  NA 8.624338624 NA 0  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA 0.002089864 0  7.803468208 NA
```

Chromosome 3 Input File (Shard 1): TS_filtering_examples/multiple_file_chr_shards/main_dir_input/Transmission_Summary_Flat_chr3_00000_annotated_for_analysis.db 
```
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
3  60082 NA G  A  NA NA NA NA NA NA NA NA NA NA intergenic_variant   NA ABCDEF   ABCDEF100   1  uncertain   PASS;ABHet=0.556;AC=2;AF=2.169e-04;AN=9220;BaseQRankSum=-2.890e-01;DP=58911;FS=0.000;GC=38.10;GQ_MEAN=37.95;GQ_STDDEV=12.17;InbreedingCoeff=-0.0017;MLEAC=2;MLEAF=2.169e-04;MQ=50.28;MQ0=0;MQRankSum=1.84;NCC=0;NDA=1;QD=6.89;ReadPosRankSum=0.361;SOR=1.329;VQSLOD=-1.253e+01;VariantType=SNP;culprit=MQ;cytoBand=3p26.3;Func=intergenic;Gene=NONE,AY269186.1;GeneDetail=NONE,dist=5349;genomicSuperDups=0.95747,chr3:75279192;FATHMM_c=0.07045;FATHMM_nc=0.00765;CSQ=A|intergenic_variant|MODIFIER||||||||||||||||||SNV||||||||||||||||||||||||||||||||||||||||||||   0/0   .:7,0:7:21:0,21,224  GT:AB:AD:DP:GQ:PL 4.92E-05 0.00026455  NA 0  NA 0  NA 5  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA 4.92E-05
3  60082 NA G  A  NA NA NA NA NA NA NA NA NA NA intergenic_variant   NA ABCDEF   ABCDEF101   1  uncertain   PASS;ABHet=0.556;AC=2;AF=2.169e-04;AN=9220;BaseQRankSum=-2.890e-01;DP=58911;FS=0.000;GC=38.10;GQ_MEAN=37.95;GQ_STDDEV=12.17;InbreedingCoeff=-0.0017;MLEAC=2;MLEAF=2.169e-04;MQ=50.28;MQ0=0;MQRankSum=1.84;NCC=0;NDA=1;QD=6.89;ReadPosRankSum=0.361;SOR=1.329;VQSLOD=-1.253e+01;VariantType=SNP;culprit=MQ;cytoBand=3p26.3;Func=intergenic;Gene=NONE,AY269186.1;GeneDetail=NONE,dist=5349;genomicSuperDups=0.95747,chr3:75279192;FATHMM_c=0.07045;FATHMM_nc=0.00765;CSQ=A|intergenic_variant|MODIFIER||||||||||||||||||SNV||||||||||||||||||||||||||||||||||||||||||||   0/0   .:6,0:6:18:0,18,193  GT:AB:AD:DP:GQ:PL 4.92E-05 0.00026455  NA 0  NA 0  NA 5  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA 4.92E-05
```
Chromosome 3 Input File (Shard 2): TS_filtering_examples/multiple_file_chr_shards/main_dir_input/Transmission_Summary_Flat_chr3_00001_annotated_for_analysis.db 
```
3  375758   NA A  G  NA NA 0.871805 0.935 NA NA NA NA NA NA intron_variant CHL1  ABCDEF   ABCDEF100   1  uncertain   PASS;ABHet=0.498;ABHom=1.000;AC=7953;AF=0.863;AN=9220;BaseQRankSum=0.950;DP=169494;FS=0.000;GC=47.62;GQ_MEAN=171.22;GQ_STDDEV=129.85;InbreedingCoeff=0.0383;MLEAC=7953;MLEAF=0.863;MQ=60.00;MQ0=0;MQRankSum=0.091;NCC=0;NDA=2;OND=3.687e-04;POSITIVE_TRAIN_SITE;QD=27.55;ReadPosRankSum=0.446;SOR=0.728;VQSLOD=23.30;VariantType=SNP;culprit=MQ;cytoBand=3p26.3;Func=intronic;Gene=CHL1;FATHMM_c=0.02636;FATHMM_nc=0.00031;GWAVA_region=0.23;GWAVA_tss=0.07;GWAVA_unmatched=182;cg46=0.935;1000g2015aug_all=0.871805;CSQ=G|intron_variant|MODIFIER|CHL1|ENSG00000134121|Transcript|ENST00000256509|protein_coding||5/27|ENST00000256509.2:c.385+5721A>G|||||||rs9858162||1|SNV|HGNC|1939|YES|||CCDS2556.1|ENSP00000256509|CHL1_HUMAN|C9JW79_HUMAN&C9JH37_HUMAN&C9JEY3_HUMAN&C9J905_HUMAN|UPI000013CF0F|1|||||A:0.1282|G:0.8192|G:0.9049|G:0.9911|G:0.836|G:0.8333|||||||||||||||||||||||  1-Jan .:0,29:29:86:.:.:891,86,0  GT:AB:AD:DP:GQ:PGT:PID:PL  0.935 0.861111111 0.709762533 0.746031746 0  0  NA 0  CHL1  NA 41.92 9.51  23.3  44.85 -1.853586132   0.0003275   0.999647383 2.51E-05 0.133650262 chr3:238279-451090   NA NA NA 0.861271676 0.738921002 0  NA
3  375758   NA A  G  NA NA 0.871805 0.935 NA NA NA NA NA NA intron_variant CHL1  ABCDEF   ABCDEF101   1  uncertain   PASS;ABHet=0.498;ABHom=1.000;AC=7953;AF=0.863;AN=9220;BaseQRankSum=0.950;DP=169494;FS=0.000;GC=47.62;GQ_MEAN=171.22;GQ_STDDEV=129.85;InbreedingCoeff=0.0383;MLEAC=7953;MLEAF=0.863;MQ=60.00;MQ0=0;MQRankSum=0.091;NCC=0;NDA=2;OND=3.687e-04;POSITIVE_TRAIN_SITE;QD=27.55;ReadPosRankSum=0.446;SOR=0.728;VQSLOD=23.30;VariantType=SNP;culprit=MQ;cytoBand=3p26.3;Func=intronic;Gene=CHL1;FATHMM_c=0.02636;FATHMM_nc=0.00031;GWAVA_region=0.23;GWAVA_tss=0.07;GWAVA_unmatched=182;cg46=0.935;1000g2015aug_all=0.871805;CSQ=G|intron_variant|MODIFIER|CHL1|ENSG00000134121|Transcript|ENST00000256509|protein_coding||5/27|ENST00000256509.2:c.385+5721A>G|||||||rs9858162||1|SNV|HGNC|1939|YES|||CCDS2556.1|ENSP00000256509|CHL1_HUMAN|C9JW79_HUMAN&C9JH37_HUMAN&C9JEY3_HUMAN&C9J905_HUMAN|UPI000013CF0F|1|||||A:0.1282|G:0.8192|G:0.9049|G:0.9911|G:0.836|G:0.8333|||||||||||||||||||||||  1-Jan .:0,24:24:72:.:.:783,72,0  GT:AB:AD:DP:GQ:PGT:PID:PL  9.35E-01 0.861111111 0.709762533 0.746031746 0  0  NA 0  CHL1  NA 41.92 9.51  23.3  44.85 -1.853586132   0.0003275   0.999647383 2.51E-05 0.133650262 chr3:238279-451090   NA NA NA 0.861271676 0.738921002 0  NA
```

**Merge All Chromosome Command for input_dir Input**:
```
bash /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/Filtering_Wrapper_Script.sh --input_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/main_dir_input --input_prefix Transmission_Summary_Flat_chr --input_suffix _annotated_for_analysis.db --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/merge_all_chr_output --input_type chr_shards --flat_or_family flat --comb_and_clean merge_all_chr --consequence upstream_gene_variant
```

* Note: Same command as that for files within input_dir chromosome subdirectories.

**Merge All Chromosome Output File**: TS_filtering_examples/multiple_file_chr_shards/merge_all_chr_output/Transmission_Summary_Flat_chr_MergedAllChr_annotated_for_analysis_filtered.db
```
#bmap command: /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/filter_flat_annotated_transmission_summary.py --input_db /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/main_dir_input/Transmission_Summary_Flat_chr1_00000_annotated_for_analysis.db --input_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/main_dir_input --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_chr_shards/merge_all_chr_output/1 --comb_and_clean no_comb_clean --input_prefix Transmission_Summary_Flat_chr --input_suffix _annotated_for_analysis.db --input_type chr_shards --flat_or_family flat --consequence upstream_gene_variant
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
1  10492 NA C  T  NA NA NA NA 0.99208  10.62 NA NA NA NA upstream_gene_variant   DDX11L1  ABCDEF   ABCDEF100   1  uncertain   PASS;ABHet=0.734;ABHom=0.953;AC=1095;AF=0.119;AN=9170;BaseQRankSum=-9.500e-01;DP=151995;FS=1.412;GC=95.24;GQ_MEAN=56.05;GQ_STDDEV=42.80;InbreedingCoeff=-0.1477;MLEAC=1158;MLEAF=0.126;MQ=53.11;MQ0=0;MQRankSum=0.043;NCC=25;NDA=1;OND=5.382e-03;QD=3.83;ReadPosRankSum=0.00;SOR=0.445;VQSLOD=-5.762e+00;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=NONE,DDX11L1;GeneDetail=NONE,dist=1377;genomicSuperDups=0.993729,chr9:10843;CADD=0.992080;CADD_Phred=10.62;FATHMM_c=0.73423;FATHMM_nc=0.05648;GWAVA_region=0.4;GWAVA_tss=0.26;GWAVA_unmatched=182;CSQ=T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000456328|processed_transcript||||||||||rs55998931|1377|1|SNV|HGNC|37102|YES|||||||||||||||||||||||||||||||||||||||||  0/1   0.330:3,6:9:67:.:.:162,0,67   GT:AB:AD:DP:GQ:PGT:PID:PL  0.0811632   0.120671284 0  0.004794885 0.527704485 0.687830688 NA 1  DDX11L1  ambiguous_previous   NA NA NA NA NA NA NA NA NA NA NA NA NA 0.147540984 0.00192864  0.096339114 8.12E-02
1  10492 NA C  T  NA NA NA NA 0.99208  10.62 NA NA NA NA upstream_gene_variant   DDX11L1  ABCDEF   ABCDEF101   1  uncertain   PASS;ABHet=0.734;ABHom=0.953;AC=1095;AF=0.119;AN=9170;BaseQRankSum=-9.500e-01;DP=151995;FS=1.412;GC=95.24;GQ_MEAN=56.05;GQ_STDDEV=42.80;InbreedingCoeff=-0.1477;MLEAC=1158;MLEAF=0.126;MQ=53.11;MQ0=0;MQRankSum=0.043;NCC=25;NDA=1;OND=5.382e-03;QD=3.83;ReadPosRankSum=0.00;SOR=0.445;VQSLOD=-5.762e+00;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=NONE,DDX11L1;GeneDetail=NONE,dist=1377;genomicSuperDups=0.993729,chr9:10843;CADD=0.992080;CADD_Phred=10.62;FATHMM_c=0.73423;FATHMM_nc=0.05648;GWAVA_region=0.4;GWAVA_tss=0.26;GWAVA_unmatched=182;CSQ=T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000456328|processed_transcript||||||||||rs55998931|1377|1|SNV|HGNC|37102|YES|||||||||||||||||||||||||||||||||||||||||  0/0   .:23,0:23:0:.:.:0,0,36  GT:AB:AD:DP:GQ:PGT:PID:PL  0.0811632   0.120671284 0  0.004794885 0.527704485 0.687830688 NA 1  DDX11L1  ambiguous_previous   NA NA NA NA NA NA NA NA NA NA NA NA NA 0.147540984 0.00192864  0.096339114 8.12E-02
```

#### Individual File Example (multi_chr)
---
**General Command**:
```
bash Filtering_Wrapper_Script.sh --input_db <input_db> --output_dir <output_dir> --input_type multi_chr --flat_or_family <flat_or_family (e.g. flat, family)> [--comb_and_clean <comb_and_clean> (e.g. merge_all_chr, merge_by_chr, or no_comb_clean)] [--no_qsub <no_qsub (e.g. True, False)>] [--log_dir <log_dir>] [--exclude_chr <exclude_chr>]
```

**Input File**: TS_filtering_examples/single_file_multi_chr/Transmission_Summary_Flat_multi_chr_annotated_for_analysis.db
```
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
1  10492 NA C  T  NA NA NA NA 0.99208  10.62 NA NA NA NA upstream_gene_variant   DDX11L1  ABCDEF   ABCDEF100   1  uncertain   PASS;ABHet=0.734;ABHom=0.953;AC=1095;AF=0.119;AN=9170;BaseQRankSum=-9.500e-01;DP=151995;FS=1.412;GC=95.24;GQ_MEAN=56.05;GQ_STDDEV=42.80;InbreedingCoeff=-0.1477;MLEAC=1158;MLEAF=0.126;MQ=53.11;MQ0=0;MQRankSum=0.043;NCC=25;NDA=1;OND=5.382e-03;QD=3.83;ReadPosRankSum=0.00;SOR=0.445;VQSLOD=-5.762e+00;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=NONE,DDX11L1;GeneDetail=NONE,dist=1377;genomicSuperDups=0.993729,chr9:10843;CADD=0.992080;CADD_Phred=10.62;FATHMM_c=0.73423;FATHMM_nc=0.05648;GWAVA_region=0.4;GWAVA_tss=0.26;GWAVA_unmatched=182;CSQ=T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000456328|processed_transcript||||||||||rs55998931|1377|1|SNV|HGNC|37102|YES|||||||||||||||||||||||||||||||||||||||||  0/0   .:23,0:23:0:.:.:0,0,36  GT:AB:AD:DP:GQ:PGT:PID:PL  0.0811632   0.120671284 0  0.004794885 0.527704485 0.687830688 NA 1  DDX11L1  ambiguous_previous   NA NA NA NA NA NA NA NA NA NA NA NA NA 0.147540984 0.00192864  0.096339114 8.12E-02
2  10205 NA T  G  NA NA NA NA NA NA NA NA NA NA regulatory_region_variant  NA ABCDEF   ABCDEF101   1  missing  PASS;ABHet=0.500;ABHom=NaN;AC=5;AF=1.132e-03;AN=4416;BaseQRankSum=0.804;DP=12674;FS=0.000;GC=61.90;GQ_MEAN=4.07;GQ_STDDEV=4.97;InbreedingCoeff=-0.1347;MLEAC=3;MLEAF=6.793e-04;MQ=59.67;MQ0=0;MQRankSum=0.804;NCC=2402;NDA=2;NEGATIVE_TRAIN_SITE;QD=34.24;ReadPosRankSum=0.804;SOR=0.538;VQSLOD=-1.159e+01;VariantType=SNP;culprit=DP;cytoBand=2p25.3;Func=intergenic;Gene=NONE,FAM110C;GeneDetail=NONE,dist=28609;FATHMM_c=0.21254;FATHMM_nc=0.36333;CSQ=G|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000675422|TF_binding_site||||||||||rs111214705|||SNV||||||||||||||||||||||||||||||||||||||||||||  ./.   .:0,0:0  GT:AB:AD:DP:GQ:PL 0  0.001088139 NA 0.001088139 NA 51.37566138 NA 2  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA
3  60082 NA G  A  NA NA NA NA NA NA NA NA NA NA intergenic_variant   NA ABCDEF   ABCDEF102   1  uncertain   PASS;ABHet=0.556;AC=2;AF=2.169e-04;AN=9220;BaseQRankSum=-2.890e-01;DP=58911;FS=0.000;GC=38.10;GQ_MEAN=37.95;GQ_STDDEV=12.17;InbreedingCoeff=-0.0017;MLEAC=2;MLEAF=2.169e-04;MQ=50.28;MQ0=0;MQRankSum=1.84;NCC=0;NDA=1;QD=6.89;ReadPosRankSum=0.361;SOR=1.329;VQSLOD=-1.253e+01;VariantType=SNP;culprit=MQ;cytoBand=3p26.3;Func=intergenic;Gene=NONE,AY269186.1;GeneDetail=NONE,dist=5349;genomicSuperDups=0.95747,chr3:75279192;FATHMM_c=0.07045;FATHMM_nc=0.00765;CSQ=A|intergenic_variant|MODIFIER||||||||||||||||||SNV||||||||||||||||||||||||||||||||||||||||||||   0/0   .:6,0:6:18:0,18,193  GT:AB:AD:DP:GQ:PL 4.92E-05 0.00026455  NA 0  NA 0  NA 5  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA 4.92E-05
```

**Merge All Chromosome Command**:
```
bash /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/Filtering_Wrapper_Script.sh --input_db /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/single_file_multi_chr/Transmission_Summary_Flat_multi_chr_annotated_for_analysis.db --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/single_file_multi_chr/merge_all_chr_output --input_type multi_chr --flat_or_family flat --inheritance_type uncertain --comb_and_clean merge_all_chr
```

**Merge All Chromosome Output File**: TS_filtering_examples/single_file_multi_chr/merge_all_chr_output/Transmission_Summary_Flat_multi_chr_annotated_for_analysis_chr_MergedAllChr_filtered.db
```
#bmap command: /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/filter_flat_annotated_transmission_summary.py --input_type per_chr --no_qsub False --inheritance_type uncertain --comb_and_clean no_comb_clean --input_db /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/single_file_multi_chr/flat_per_chr_shards/1/Transmission_Summary_Flat_multi_chr_annotated_for_analysis_chr1_00000.gz --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/single_file_multi_chr/1 --flat_or_family flat
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
1  10492 NA C  T  NA NA NA NA 0.99208  10.62 NA NA NA NA upstream_gene_variant   DDX11L1  ABCDEF   ABCDEF100   1  uncertain   PASS;ABHet=0.734;ABHom=0.953;AC=1095;AF=0.119;AN=9170;BaseQRankSum=-9.500e-01;DP=151995;FS=1.412;GC=95.24;GQ_MEAN=56.05;GQ_STDDEV=42.80;InbreedingCoeff=-0.1477;MLEAC=1158;MLEAF=0.126;MQ=53.11;MQ0=0;MQRankSum=0.043;NCC=25;NDA=1;OND=5.382e-03;QD=3.83;ReadPosRankSum=0.00;SOR=0.445;VQSLOD=-5.762e+00;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=NONE,DDX11L1;GeneDetail=NONE,dist=1377;genomicSuperDups=0.993729,chr9:10843;CADD=0.992080;CADD_Phred=10.62;FATHMM_c=0.73423;FATHMM_nc=0.05648;GWAVA_region=0.4;GWAVA_tss=0.26;GWAVA_unmatched=182;CSQ=T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000456328|processed_transcript||||||||||rs55998931|1377|1|SNV|HGNC|37102|YES|||||||||||||||||||||||||||||||||||||||||  0/0   .:23,0:23:0:.:.:0,0,36  GT:AB:AD:DP:GQ:PGT:PID:PL  0.0811632   0.120671284 0  0.004794885 0.527704485 0.687830688 NA 1  DDX11L1  ambiguous_previous   NA NA NA NA NA NA NA NA NA NA NA NA NA 0.147540984 0.00192864  0.096339114 8.12E-02
3  60082 NA G  A  NA NA NA NA NA NA NA NA NA NA intergenic_variant   NA ABCDEF   ABCDEF102   1  uncertain   PASS;ABHet=0.556;AC=2;AF=2.169e-04;AN=9220;BaseQRankSum=-2.890e-01;DP=58911;FS=0.000;GC=38.10;GQ_MEAN=37.95;GQ_STDDEV=12.17;InbreedingCoeff=-0.0017;MLEAC=2;MLEAF=2.169e-04;MQ=50.28;MQ0=0;MQRankSum=1.84;NCC=0;NDA=1;QD=6.89;ReadPosRankSum=0.361;SOR=1.329;VQSLOD=-1.253e+01;VariantType=SNP;culprit=MQ;cytoBand=3p26.3;Func=intergenic;Gene=NONE,AY269186.1;GeneDetail=NONE,dist=5349;genomicSuperDups=0.95747,chr3:75279192;FATHMM_c=0.07045;FATHMM_nc=0.00765;CSQ=A|intergenic_variant|MODIFIER||||||||||||||||||SNV||||||||||||||||||||||||||||||||||||||||||||   0/0   .:6,0:6:18:0,18,193  GT:AB:AD:DP:GQ:PL 4.92E-05 0.00026455  NA 0  NA 0  NA 5  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA 4.92E-05
```

**Merge by Chromosome Command**:
```
bash /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/Filtering_Wrapper_Script.sh --input_db /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/single_file_multi_chr/Transmission_Summary_Flat_multi_chr_annotated_for_analysis.db --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/single_file_multi_chr/merge_by_chr_output --input_type multi_chr --flat_or_family flat --inheritance_type uncertain --comb_and_clean merge_by_chr
```

**Merge by Chromosome Output Files**:

Chromosome 1 Output File: TS_filtering_examples/single_file_multi_chr/merge_by_chr_output/Transmission_Summary_Flat_multi_chr_annotated_for_analysis_chr1_filtered.db
```
#bmap command: /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/filter_flat_annotated_transmission_summary.py --input_type per_chr --no_qsub False --inheritance_type uncertain --comb_and_clean no_comb_clean --input_db /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/flat_per_chr_shards/1/Transmission_Summary_Flat_multi_chr_annotated_for_analysis_chr1_00000.gz --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/single_file_multi_chr/1 --flat_or_family flat
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
1  10492 NA C  T  NA NA NA NA 0.99208  10.62 NA NA NA NA upstream_gene_variant   DDX11L1  ABCDEF   ABCDEF100   1  uncertain   PASS;ABHet=0.734;ABHom=0.953;AC=1095;AF=0.119;AN=9170;BaseQRankSum=-9.500e-01;DP=151995;FS=1.412;GC=95.24;GQ_MEAN=56.05;GQ_STDDEV=42.80;InbreedingCoeff=-0.1477;MLEAC=1158;MLEAF=0.126;MQ=53.11;MQ0=0;MQRankSum=0.043;NCC=25;NDA=1;OND=5.382e-03;QD=3.83;ReadPosRankSum=0.00;SOR=0.445;VQSLOD=-5.762e+00;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=NONE,DDX11L1;GeneDetail=NONE,dist=1377;genomicSuperDups=0.993729,chr9:10843;CADD=0.992080;CADD_Phred=10.62;FATHMM_c=0.73423;FATHMM_nc=0.05648;GWAVA_region=0.4;GWAVA_tss=0.26;GWAVA_unmatched=182;CSQ=T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000456328|processed_transcript||||||||||rs55998931|1377|1|SNV|HGNC|37102|YES|||||||||||||||||||||||||||||||||||||||||  0/0   .:23,0:23:0:.:.:0,0,36  GT:AB:AD:DP:GQ:PGT:PID:PL  0.0811632   0.120671284 0  0.004794885 0.527704485 0.687830688 NA 1  DDX11L1  ambiguous_previous   NA NA NA NA NA NA NA NA NA NA NA NA NA 0.147540984 0.00192864  0.096339114 8.12E-02
```

Chromosme 2 Output File: TS_filtering_examples/single_file_multi_chr/merge_by_chr_output/Transmission_Summary_Flat_multi_chr_annotated_for_analysis_chr2_filtered.db
```
Empty file, since no variants met filtering requirements.
```

Chromosome 3 Output File: TS_filtering_examples/single_file_multi_chr/merge_by_chr_output/Transmission_Summary_Flat_multi_chr_annotated_for_analysis_chr3_filtered.db
```
3  60082 NA G  A  NA NA NA NA NA NA NA NA NA NA intergenic_variant   NA ABCDEF   ABCDEF102   1  uncertain   PASS;ABHet=0.556;AC=2;AF=2.169e-04;AN=9220;BaseQRankSum=-2.890e-01;DP=58911;FS=0.000;GC=38.10;GQ_MEAN=37.95;GQ_STDDEV=12.17;InbreedingCoeff=-0.0017;MLEAC=2;MLEAF=2.169e-04;MQ=50.28;MQ0=0;MQRankSum=1.84;NCC=0;NDA=1;QD=6.89;ReadPosRankSum=0.361;SOR=1.329;VQSLOD=-1.253e+01;VariantType=SNP;culprit=MQ;cytoBand=3p26.3;Func=intergenic;Gene=NONE,AY269186.1;GeneDetail=NONE,dist=5349;genomicSuperDups=0.95747,chr3:75279192;FATHMM_c=0.07045;FATHMM_nc=0.00765;CSQ=A|intergenic_variant|MODIFIER||||||||||||||||||SNV||||||||||||||||||||||||||||||||||||||||||||   0/0   .:6,0:6:18:0,18,193  GT:AB:AD:DP:GQ:PL 4.92E-05 0.00026455  NA 0  NA 0  NA 5  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA 4.92E-05
```

**No Combine and Clean Command**:
```
bash /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/Filtering_Wrapper_Script.sh --input_db /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/single_file_multi_chr/Transmission_Summary_Flat_multi_chr_annotated_for_analysis.db --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/single_file_multi_chr/no_comb_clean_output --input_type multi_chr --flat_or_family flat --inheritance_type uncertain --comb_and_clean no_comb_clean
```

**No Combine and Clean Output Files**:
Chromosome 1 Output File: TS_filtering_examples/single_file_multi_chr/no_comb_clean_output/1/Transmission_Summary_Flat_multi_chr_annotated_for_analysis_chr1_00000_filtered.db
```
#bmap command: /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/filter_flat_annotated_transmission_summary.py --input_type per_chr --no_qsub False --inheritance_type uncertain --comb_and_clean no_comb_clean --input_db /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/flat_per_chr_shards/1/Transmission_Summary_Flat_multi_chr_annotated_for_analysis_chr1_00000.gz --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/single_file_multi_chr/1 --flat_or_family flat
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
1  10492 NA C  T  NA NA NA NA 0.99208  10.62 NA NA NA NA upstream_gene_variant   DDX11L1  ABCDEF   ABCDEF100   1  uncertain   PASS;ABHet=0.734;ABHom=0.953;AC=1095;AF=0.119;AN=9170;BaseQRankSum=-9.500e-01;DP=151995;FS=1.412;GC=95.24;GQ_MEAN=56.05;GQ_STDDEV=42.80;InbreedingCoeff=-0.1477;MLEAC=1158;MLEAF=0.126;MQ=53.11;MQ0=0;MQRankSum=0.043;NCC=25;NDA=1;OND=5.382e-03;QD=3.83;ReadPosRankSum=0.00;SOR=0.445;VQSLOD=-5.762e+00;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=NONE,DDX11L1;GeneDetail=NONE,dist=1377;genomicSuperDups=0.993729,chr9:10843;CADD=0.992080;CADD_Phred=10.62;FATHMM_c=0.73423;FATHMM_nc=0.05648;GWAVA_region=0.4;GWAVA_tss=0.26;GWAVA_unmatched=182;CSQ=T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000456328|processed_transcript||||||||||rs55998931|1377|1|SNV|HGNC|37102|YES|||||||||||||||||||||||||||||||||||||||||  0/0   .:23,0:23:0:.:.:0,0,36  GT:AB:AD:DP:GQ:PGT:PID:PL  0.0811632   0.120671284 0  0.004794885 0.527704485 0.687830688 NA 1  DDX11L1  ambiguous_previous   NA NA NA NA NA NA NA NA NA NA NA NA NA 0.147540984 0.00192864  0.096339114 8.12E-02
```

Chromosme 2 Output File: TS_filtering_examples/single_file_multi_chr/no_comb_clean_output/2/Transmission_Summary_Flat_multi_chr_annotated_for_analysis_chr2_00000_filtered.db
```
Empty file, since no variants met filtering requirements.
```

Chromosome 3 Output File: TS_filtering_examples/single_file_multi_chr/no_comb_clean_output/3/Transmission_Summary_Flat_multi_chr_annotated_for_analysis_chr3_00000_filtered.db
```
3  60082 NA G  A  NA NA NA NA NA NA NA NA NA NA intergenic_variant   NA ABCDEF   ABCDEF102   1  uncertain   PASS;ABHet=0.556;AC=2;AF=2.169e-04;AN=9220;BaseQRankSum=-2.890e-01;DP=58911;FS=0.000;GC=38.10;GQ_MEAN=37.95;GQ_STDDEV=12.17;InbreedingCoeff=-0.0017;MLEAC=2;MLEAF=2.169e-04;MQ=50.28;MQ0=0;MQRankSum=1.84;NCC=0;NDA=1;QD=6.89;ReadPosRankSum=0.361;SOR=1.329;VQSLOD=-1.253e+01;VariantType=SNP;culprit=MQ;cytoBand=3p26.3;Func=intergenic;Gene=NONE,AY269186.1;GeneDetail=NONE,dist=5349;genomicSuperDups=0.95747,chr3:75279192;FATHMM_c=0.07045;FATHMM_nc=0.00765;CSQ=A|intergenic_variant|MODIFIER||||||||||||||||||SNV||||||||||||||||||||||||||||||||||||||||||||   0/0   .:6,0:6:18:0,18,193  GT:AB:AD:DP:GQ:PL 4.92E-05 0.00026455  NA 0  NA 0  NA 5  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA 4.92E-05
```

#### Multiple Files Example (per_chr)
---
**General Command**:
```
bash Filtering_Wrapper_Script.sh --input_dir <input_dir> --input_prefix <input_prefix> --input_suffix <input_suffix> --output_dir <output_dir> --input_type per_chr --flat_or_family <flat_or_family (e.g. flat, family)> [--comb_and_clean <comb_and_clean> (e.g. merge_all_chr, merge_by_chr, or no_comb_clean)] [--no_qsub <no_qsub (e.g. True, False)>] [--log_dir <log_dir>] [--exclude_chr <exclude_chr>]
```
**Input Files Within Provided input_dir**: <br/>
Chromosome 1 Input File: TS_filtering_examples/multiple_file_per_chr/main_dir_input/Transmission_Summary_Flat_chr1_annotated_for_analysis.db
```
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
1  10492 NA C  T  NA NA NA NA 0.99208  10.62 NA NA NA NA upstream_gene_variant   DDX11L1  ABCDEF   ABCDEF100   1  uncertain   PASS;ABHet=0.734;ABHom=0.953;AC=1095;AF=0.119;AN=9170;BaseQRankSum=-9.500e-01;DP=151995;FS=1.412;GC=95.24;GQ_MEAN=56.05;GQ_STDDEV=42.80;InbreedingCoeff=-0.1477;MLEAC=1158;MLEAF=0.126;MQ=53.11;MQ0=0;MQRankSum=0.043;NCC=25;NDA=1;OND=5.382e-03;QD=3.83;ReadPosRankSum=0.00;SOR=0.445;VQSLOD=-5.762e+00;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=NONE,DDX11L1;GeneDetail=NONE,dist=1377;genomicSuperDups=0.993729,chr9:10843;CADD=0.992080;CADD_Phred=10.62;FATHMM_c=0.73423;FATHMM_nc=0.05648;GWAVA_region=0.4;GWAVA_tss=0.26;GWAVA_unmatched=182;CSQ=T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000456328|processed_transcript||||||||||rs55998931|1377|1|SNV|HGNC|37102|YES|||||||||||||||||||||||||||||||||||||||||  0/1   0.330:3,6:9:67:.:.:162,0,67   GT:AB:AD:DP:GQ:PGT:PID:PL  0.0811632   0.120671284 0  0.004794885 0.527704485 0.687830688 NA 1  DDX11L1  ambiguous_previous   NA NA NA NA NA NA NA NA NA NA NA NA NA 0.147540984 0.00192864  0.096339114 8.12E-02
1  10492 NA C  T  NA NA NA NA 0.99208  10.62 NA NA NA NA upstream_gene_variant   DDX11L1  ABCDEF   ABCDEF101   1  uncertain   PASS;ABHet=0.734;ABHom=0.953;AC=1095;AF=0.119;AN=9170;BaseQRankSum=-9.500e-01;DP=151995;FS=1.412;GC=95.24;GQ_MEAN=56.05;GQ_STDDEV=42.80;InbreedingCoeff=-0.1477;MLEAC=1158;MLEAF=0.126;MQ=53.11;MQ0=0;MQRankSum=0.043;NCC=25;NDA=1;OND=5.382e-03;QD=3.83;ReadPosRankSum=0.00;SOR=0.445;VQSLOD=-5.762e+00;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=NONE,DDX11L1;GeneDetail=NONE,dist=1377;genomicSuperDups=0.993729,chr9:10843;CADD=0.992080;CADD_Phred=10.62;FATHMM_c=0.73423;FATHMM_nc=0.05648;GWAVA_region=0.4;GWAVA_tss=0.26;GWAVA_unmatched=182;CSQ=T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000456328|processed_transcript||||||||||rs55998931|1377|1|SNV|HGNC|37102|YES|||||||||||||||||||||||||||||||||||||||||  0/0   .:23,0:23:0:.:.:0,0,36  GT:AB:AD:DP:GQ:PGT:PID:PL  0.0811632   0.120671284 0  0.004794885 0.527704485 0.687830688 NA 1  DDX11L1  ambiguous_previous   NA NA NA NA NA NA NA NA NA NA NA NA NA 0.147540984 0.00192864  0.096339114 8.12E-02
```
Chromsome 2 Input File: TS_filtering_examples/multiple_file_per_chr/main_dir_input/Transmission_Summary_Flat_chr2_annotated_for_analysis.db
```
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
2  10205 NA T  G  NA NA NA NA NA NA NA NA NA NA regulatory_region_variant  NA ABCDEF   ABCDEF101   0  uncertain   PASS;ABHet=0.500;ABHom=NaN;AC=5;AF=1.132e-03;AN=4416;BaseQRankSum=0.804;DP=12674;FS=0.000;GC=61.90;GQ_MEAN=4.07;GQ_STDDEV=4.97;InbreedingCoeff=-0.1347;MLEAC=3;MLEAF=6.793e-04;MQ=59.67;MQ0=0;MQRankSum=0.804;NCC=2402;NDA=2;NEGATIVE_TRAIN_SITE;QD=34.24;ReadPosRankSum=0.804;SOR=0.538;VQSLOD=-1.159e+01;VariantType=SNP;culprit=DP;cytoBand=2p25.3;Func=intergenic;Gene=NONE,FAM110C;GeneDetail=NONE,dist=28609;FATHMM_c=0.21254;FATHMM_nc=0.36333;CSQ=G|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000675422|TF_binding_site||||||||||rs111214705|||SNV||||||||||||||||||||||||||||||||||||||||||||  0/0   .:3,0:3:0:0,0,40  GT:AB:AD:DP:GQ:PL 0  0.001088139 NA 0.001088139 NA 51.37566138 NA 2  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA
2  10205 NA T  G  NA NA NA NA NA NA NA NA NA NA regulatory_region_variant  NA ABCDEF   ABCDEF102   1  missing  PASS;ABHet=0.500;ABHom=NaN;AC=5;AF=1.132e-03;AN=4416;BaseQRankSum=0.804;DP=12674;FS=0.000;GC=61.90;GQ_MEAN=4.07;GQ_STDDEV=4.97;InbreedingCoeff=-0.1347;MLEAC=3;MLEAF=6.793e-04;MQ=59.67;MQ0=0;MQRankSum=0.804;NCC=2402;NDA=2;NEGATIVE_TRAIN_SITE;QD=34.24;ReadPosRankSum=0.804;SOR=0.538;VQSLOD=-1.159e+01;VariantType=SNP;culprit=DP;cytoBand=2p25.3;Func=intergenic;Gene=NONE,FAM110C;GeneDetail=NONE,dist=28609;FATHMM_c=0.21254;FATHMM_nc=0.36333;CSQ=G|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000675422|TF_binding_site||||||||||rs111214705|||SNV||||||||||||||||||||||||||||||||||||||||||||  ./.   .:0,0:0  GT:AB:AD:DP:GQ:PL 0  0.001088139 NA 0.001088139 NA 51.37566138 NA 2  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA
```

Chromosome 3 Input File: TS_filtering_examples/multiple_file_per_chr/main_dir_input/Transmission_Summary_Flat_chr3_annotated_for_analysis.db
```
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
3  60082 NA G  A  NA NA NA NA NA NA NA NA NA NA intergenic_variant   NA ABCDEF   ABCDEF101   1  uncertain   PASS;ABHet=0.556;AC=2;AF=2.169e-04;AN=9220;BaseQRankSum=-2.890e-01;DP=58911;FS=0.000;GC=38.10;GQ_MEAN=37.95;GQ_STDDEV=12.17;InbreedingCoeff=-0.0017;MLEAC=2;MLEAF=2.169e-04;MQ=50.28;MQ0=0;MQRankSum=1.84;NCC=0;NDA=1;QD=6.89;ReadPosRankSum=0.361;SOR=1.329;VQSLOD=-1.253e+01;VariantType=SNP;culprit=MQ;cytoBand=3p26.3;Func=intergenic;Gene=NONE,AY269186.1;GeneDetail=NONE,dist=5349;genomicSuperDups=0.95747,chr3:75279192;FATHMM_c=0.07045;FATHMM_nc=0.00765;CSQ=A|intergenic_variant|MODIFIER||||||||||||||||||SNV||||||||||||||||||||||||||||||||||||||||||||   0/0   .:7,0:7:21:0,21,224  GT:AB:AD:DP:GQ:PL 4.92E-05 0.00026455  NA 0  NA 0  NA 5  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA 4.92E-05
3  60082 NA G  A  NA NA NA NA NA NA NA NA NA NA intergenic_variant   NA ABCDEF   ABCDEF102   1  uncertain   PASS;ABHet=0.556;AC=2;AF=2.169e-04;AN=9220;BaseQRankSum=-2.890e-01;DP=58911;FS=0.000;GC=38.10;GQ_MEAN=37.95;GQ_STDDEV=12.17;InbreedingCoeff=-0.0017;MLEAC=2;MLEAF=2.169e-04;MQ=50.28;MQ0=0;MQRankSum=1.84;NCC=0;NDA=1;QD=6.89;ReadPosRankSum=0.361;SOR=1.329;VQSLOD=-1.253e+01;VariantType=SNP;culprit=MQ;cytoBand=3p26.3;Func=intergenic;Gene=NONE,AY269186.1;GeneDetail=NONE,dist=5349;genomicSuperDups=0.95747,chr3:75279192;FATHMM_c=0.07045;FATHMM_nc=0.00765;CSQ=A|intergenic_variant|MODIFIER||||||||||||||||||SNV||||||||||||||||||||||||||||||||||||||||||||   0/0   .:6,0:6:18:0,18,193  GT:AB:AD:DP:GQ:PL 4.92E-05 0.00026455  NA 0  NA 0  NA 5  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA 4.92E-05
```

**Merge All Chromosome Command**:
```
bash /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/Filtering_Wrapper_Script.sh --input_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_per_chr/main_dir_input --input_prefix Transmission_Summary_Flat_chr --input_suffix _annotated_for_analysis.db --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_per_chr/merge_all_chr_output --input_type per_chr --flat_or_family flat --comb_and_clean merge_all_chr --consequence regulatory_region_variant
```

**Merge All Chromosome Output File**: TS_filtering_examples/multiple_file_per_chr/merge_all_chr_output/Transmission_Summary_Flat_chr_MergedAllChr_annotated_for_analysis_filtered.db
```
#bmap command: /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/filter_flat_annotated_transmission_summary.py --input_db /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_per_chr/main_dir_input/Transmission_Summary_Flat_chr1_annotated_for_analysis.db --comb_and_clean no_comb_clean --input_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_per_chr/main_dir_input --input_prefix Transmission_Summary_Flat_chr --input_suffix _annotated_for_analysis.db --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_per_chr/main_dir_input --input_type per_chr --flat_or_family flat --consequence regulatory_region_variant
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
2  10205 NA T  G  NA NA NA NA NA NA NA NA NA NA regulatory_region_variant  NA ABCDEF   ABCDEF101   0  uncertain   PASS;ABHet=0.500;ABHom=NaN;AC=5;AF=1.132e-03;AN=4416;BaseQRankSum=0.804;DP=12674;FS=0.000;GC=61.90;GQ_MEAN=4.07;GQ_STDDEV=4.97;InbreedingCoeff=-0.1347;MLEAC=3;MLEAF=6.793e-04;MQ=59.67;MQ0=0;MQRankSum=0.804;NCC=2402;NDA=2;NEGATIVE_TRAIN_SITE;QD=34.24;ReadPosRankSum=0.804;SOR=0.538;VQSLOD=-1.159e+01;VariantType=SNP;culprit=DP;cytoBand=2p25.3;Func=intergenic;Gene=NONE,FAM110C;GeneDetail=NONE,dist=28609;FATHMM_c=0.21254;FATHMM_nc=0.36333;CSQ=G|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000675422|TF_binding_site||||||||||rs111214705|||SNV||||||||||||||||||||||||||||||||||||||||||||  0/0   .:3,0:3:0:0,0,40  GT:AB:AD:DP:GQ:PL 0  0.001088139 NA 0.001088139 NA 51.37566138 NA 2  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA
2  10205 NA T  G  NA NA NA NA NA NA NA NA NA NA regulatory_region_variant  NA ABCDEF   ABCDEF102   1  missing  PASS;ABHet=0.500;ABHom=NaN;AC=5;AF=1.132e-03;AN=4416;BaseQRankSum=0.804;DP=12674;FS=0.000;GC=61.90;GQ_MEAN=4.07;GQ_STDDEV=4.97;InbreedingCoeff=-0.1347;MLEAC=3;MLEAF=6.793e-04;MQ=59.67;MQ0=0;MQRankSum=0.804;NCC=2402;NDA=2;NEGATIVE_TRAIN_SITE;QD=34.24;ReadPosRankSum=0.804;SOR=0.538;VQSLOD=-1.159e+01;VariantType=SNP;culprit=DP;cytoBand=2p25.3;Func=intergenic;Gene=NONE,FAM110C;GeneDetail=NONE,dist=28609;FATHMM_c=0.21254;FATHMM_nc=0.36333;CSQ=G|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000675422|TF_binding_site||||||||||rs111214705|||SNV||||||||||||||||||||||||||||||||||||||||||||  ./.   .:0,0:0  GT:AB:AD:DP:GQ:PL 0  0.001088139 NA 0.001088139 NA 51.37566138 NA 2  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA
```

**No Combine and Clean Command**:
```
bash /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/Filtering_Wrapper_Script.sh --input_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_per_chr/main_dir_input --input_prefix Transmission_Summary_Flat_chr --input_suffix _annotated_for_analysis.db --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_per_chr/merge_all_chr_output --input_type per_chr --flat_or_family flat --comb_and_clean no_comb_clean --consequence regulatory_region_variant
```

**No Combine and Clean Output**: <br/>
Chromosome 1 Output File: TS_filtering_examples/multiple_file_per_chr/no_comb_clean_output/Transmission_Summary_Flat_chr1_annotated_for_analysis_filtered.db
```
#bmap command: /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/filter_flat_annotated_transmission_summary.py --input_db /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_per_chr/main_dir_input/Transmission_Summary_Flat_chr1_annotated_for_analysis.db --comb_and_clean no_comb_clean --input_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_per_chr/main_dir_input --input_prefix Transmission_Summary_Flat_chr --input_suffix _annotated_for_analysis.db --output_dir [/ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_per_chr/no_comb_clean_output --input_type per_chr --flat_or_family flat --consequence regulatory_region_variant
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
   - Header only provided, since no variants met filtering requirements.
```
Chromosome 2 Output File: TS_filtering_examples/multiple_file_per_chr/no_comb_clean_output/Transmission_Summary_Flat_chr2_annotated_for_analysis_filtered.db
```
#bmap command: /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/filter_flat_annotated_transmission_summary.py --input_db /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_per_chr/main_dir_input/Transmission_Summary_Flat_chr2_annotated_for_analysis.db --comb_and_clean no_comb_clean --input_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_per_chr/main_dir_input --input_prefix Transmission_Summary_Flat_chr --input_suffix _annotated_for_analysis.db --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_per_chr/no_comb_clean_output --input_type per_chr --flat_or_family flat --consequence regulatory_region_variant
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
2  10205 NA T  G  NA NA NA NA NA NA NA NA NA NA regulatory_region_variant  NA ABCDEF   ABCDEF101   0  uncertain   PASS;ABHet=0.500;ABHom=NaN;AC=5;AF=1.132e-03;AN=4416;BaseQRankSum=0.804;DP=12674;FS=0.000;GC=61.90;GQ_MEAN=4.07;GQ_STDDEV=4.97;InbreedingCoeff=-0.1347;MLEAC=3;MLEAF=6.793e-04;MQ=59.67;MQ0=0;MQRankSum=0.804;NCC=2402;NDA=2;NEGATIVE_TRAIN_SITE;QD=34.24;ReadPosRankSum=0.804;SOR=0.538;VQSLOD=-1.159e+01;VariantType=SNP;culprit=DP;cytoBand=2p25.3;Func=intergenic;Gene=NONE,FAM110C;GeneDetail=NONE,dist=28609;FATHMM_c=0.21254;FATHMM_nc=0.36333;CSQ=G|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000675422|TF_binding_site||||||||||rs111214705|||SNV||||||||||||||||||||||||||||||||||||||||||||  0/0   .:3,0:3:0:0,0,40  GT:AB:AD:DP:GQ:PL 0  0.001088139 NA 0.001088139 NA 51.37566138 NA 2  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA
2  10205 NA T  G  NA NA NA NA NA NA NA NA NA NA regulatory_region_variant  NA ABCDEF   ABCDEF102   1  missing  PASS;ABHet=0.500;ABHom=NaN;AC=5;AF=1.132e-03;AN=4416;BaseQRankSum=0.804;DP=12674;FS=0.000;GC=61.90;GQ_MEAN=4.07;GQ_STDDEV=4.97;InbreedingCoeff=-0.1347;MLEAC=3;MLEAF=6.793e-04;MQ=59.67;MQ0=0;MQRankSum=0.804;NCC=2402;NDA=2;NEGATIVE_TRAIN_SITE;QD=34.24;ReadPosRankSum=0.804;SOR=0.538;VQSLOD=-1.159e+01;VariantType=SNP;culprit=DP;cytoBand=2p25.3;Func=intergenic;Gene=NONE,FAM110C;GeneDetail=NONE,dist=28609;FATHMM_c=0.21254;FATHMM_nc=0.36333;CSQ=G|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000675422|TF_binding_site||||||||||rs111214705|||SNV||||||||||||||||||||||||||||||||||||||||||||  ./.   .:0,0:0  GT:AB:AD:DP:GQ:PL 0  0.001088139 NA 0.001088139 NA 51.37566138 NA 2  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA
```

Chromosome 3 Output File: TS_filtering_examples/multiple_file_per_chr/no_comb_clean_output/Transmission_Summary_Flat_chr3_annotated_for_analysis_filtered.db
```
#bmap command: /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/filter_flat_annotated_transmission_summary.py --input_db /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_per_chr/main_dir_input/Transmission_Summary_Flat_chr3_annotated_for_analysis.db --comb_and_clean no_comb_clean --input_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_per_chr/main_dir_input --input_prefix Transmission_Summary_Flat_chr --input_suffix _annotated_for_analysis.db --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_per_chr/no_comb_clean_output --input_type per_chr --flat_or_family flat --consequence regulatory_region_variant
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
   - Header only provided, since no variants met filtering requirements.
```

* Note: No merge by chromsome command was provided, since this comb_and_clean option is not recommended for multiple, per_chr input files. If the merge_by_chr option is provided for the comb_and_clean argument, the output will be the same as the comb_and_clean output above.


**Input Files Within Subdirectory of input_dir**: <br/>

* Note: Input files can be in chromosome subdirectories within input_dir

Chromosome 1 Input File: TS_filtering_examples/multiple_file_per_chr/subdir_input/1/Transmission_Summary_Flat_chr1_annotated_for_analysis.db
```
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
1  10492 NA C  T  NA NA NA NA 0.99208  10.62 NA NA NA NA upstream_gene_variant   DDX11L1  ABCDEF   ABCDEF100   1  uncertain   PASS;ABHet=0.734;ABHom=0.953;AC=1095;AF=0.119;AN=9170;BaseQRankSum=-9.500e-01;DP=151995;FS=1.412;GC=95.24;GQ_MEAN=56.05;GQ_STDDEV=42.80;InbreedingCoeff=-0.1477;MLEAC=1158;MLEAF=0.126;MQ=53.11;MQ0=0;MQRankSum=0.043;NCC=25;NDA=1;OND=5.382e-03;QD=3.83;ReadPosRankSum=0.00;SOR=0.445;VQSLOD=-5.762e+00;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=NONE,DDX11L1;GeneDetail=NONE,dist=1377;genomicSuperDups=0.993729,chr9:10843;CADD=0.992080;CADD_Phred=10.62;FATHMM_c=0.73423;FATHMM_nc=0.05648;GWAVA_region=0.4;GWAVA_tss=0.26;GWAVA_unmatched=182;CSQ=T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000456328|processed_transcript||||||||||rs55998931|1377|1|SNV|HGNC|37102|YES|||||||||||||||||||||||||||||||||||||||||  0/1   0.330:3,6:9:67:.:.:162,0,67   GT:AB:AD:DP:GQ:PGT:PID:PL  0.0811632   0.120671284 0  0.004794885 0.527704485 0.687830688 NA 1  DDX11L1  ambiguous_previous   NA NA NA NA NA NA NA NA NA NA NA NA NA 0.147540984 0.00192864  0.096339114 8.12E-02
1  10492 NA C  T  NA NA NA NA 0.99208  10.62 NA NA NA NA upstream_gene_variant   DDX11L1  ABCDEF   ABCDEF101   1  uncertain   PASS;ABHet=0.734;ABHom=0.953;AC=1095;AF=0.119;AN=9170;BaseQRankSum=-9.500e-01;DP=151995;FS=1.412;GC=95.24;GQ_MEAN=56.05;GQ_STDDEV=42.80;InbreedingCoeff=-0.1477;MLEAC=1158;MLEAF=0.126;MQ=53.11;MQ0=0;MQRankSum=0.043;NCC=25;NDA=1;OND=5.382e-03;QD=3.83;ReadPosRankSum=0.00;SOR=0.445;VQSLOD=-5.762e+00;VariantType=SNP;culprit=MQ;cytoBand=1p36.33;Func=intergenic;Gene=NONE,DDX11L1;GeneDetail=NONE,dist=1377;genomicSuperDups=0.993729,chr9:10843;CADD=0.992080;CADD_Phred=10.62;FATHMM_c=0.73423;FATHMM_nc=0.05648;GWAVA_region=0.4;GWAVA_tss=0.26;GWAVA_unmatched=182;CSQ=T|upstream_gene_variant|MODIFIER|DDX11L1|ENSG00000223972|Transcript|ENST00000456328|processed_transcript||||||||||rs55998931|1377|1|SNV|HGNC|37102|YES|||||||||||||||||||||||||||||||||||||||||  0/0   .:23,0:23:0:.:.:0,0,36  GT:AB:AD:DP:GQ:PGT:PID:PL  0.0811632   0.120671284 0  0.004794885 0.527704485 0.687830688 NA 1  DDX11L1  ambiguous_previous   NA NA NA NA NA NA NA NA NA NA NA NA NA 0.147540984 0.00192864  0.096339114 8.12E-02
```
Chromsome 2 Input File: TS_filtering_examples/multiple_file_per_chr/subdir_input/2/Transmission_Summary_Flat_chr2_annotated_for_analysis.db
```
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
2  10205 NA T  G  NA NA NA NA NA NA NA NA NA NA regulatory_region_variant  NA ABCDEF   ABCDEF101   0  uncertain   PASS;ABHet=0.500;ABHom=NaN;AC=5;AF=1.132e-03;AN=4416;BaseQRankSum=0.804;DP=12674;FS=0.000;GC=61.90;GQ_MEAN=4.07;GQ_STDDEV=4.97;InbreedingCoeff=-0.1347;MLEAC=3;MLEAF=6.793e-04;MQ=59.67;MQ0=0;MQRankSum=0.804;NCC=2402;NDA=2;NEGATIVE_TRAIN_SITE;QD=34.24;ReadPosRankSum=0.804;SOR=0.538;VQSLOD=-1.159e+01;VariantType=SNP;culprit=DP;cytoBand=2p25.3;Func=intergenic;Gene=NONE,FAM110C;GeneDetail=NONE,dist=28609;FATHMM_c=0.21254;FATHMM_nc=0.36333;CSQ=G|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000675422|TF_binding_site||||||||||rs111214705|||SNV||||||||||||||||||||||||||||||||||||||||||||  0/0   .:3,0:3:0:0,0,40  GT:AB:AD:DP:GQ:PL 0  0.001088139 NA 0.001088139 NA 51.37566138 NA 2  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA
2  10205 NA T  G  NA NA NA NA NA NA NA NA NA NA regulatory_region_variant  NA ABCDEF   ABCDEF102   1  missing  PASS;ABHet=0.500;ABHom=NaN;AC=5;AF=1.132e-03;AN=4416;BaseQRankSum=0.804;DP=12674;FS=0.000;GC=61.90;GQ_MEAN=4.07;GQ_STDDEV=4.97;InbreedingCoeff=-0.1347;MLEAC=3;MLEAF=6.793e-04;MQ=59.67;MQ0=0;MQRankSum=0.804;NCC=2402;NDA=2;NEGATIVE_TRAIN_SITE;QD=34.24;ReadPosRankSum=0.804;SOR=0.538;VQSLOD=-1.159e+01;VariantType=SNP;culprit=DP;cytoBand=2p25.3;Func=intergenic;Gene=NONE,FAM110C;GeneDetail=NONE,dist=28609;FATHMM_c=0.21254;FATHMM_nc=0.36333;CSQ=G|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000675422|TF_binding_site||||||||||rs111214705|||SNV||||||||||||||||||||||||||||||||||||||||||||  ./.   .:0,0:0  GT:AB:AD:DP:GQ:PL 0  0.001088139 NA 0.001088139 NA 51.37566138 NA 2  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA
```
Chromosome 3 Input File: TS_filtering_examples/multiple_file_per_chr/subdir_input/2/Transmission_Summary_Flat_chr3_annotated_for_analysis.db
```
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
3  60082 NA G  A  NA NA NA NA NA NA NA NA NA NA intergenic_variant   NA ABCDEF   ABCDEF101   1  uncertain   PASS;ABHet=0.556;AC=2;AF=2.169e-04;AN=9220;BaseQRankSum=-2.890e-01;DP=58911;FS=0.000;GC=38.10;GQ_MEAN=37.95;GQ_STDDEV=12.17;InbreedingCoeff=-0.0017;MLEAC=2;MLEAF=2.169e-04;MQ=50.28;MQ0=0;MQRankSum=1.84;NCC=0;NDA=1;QD=6.89;ReadPosRankSum=0.361;SOR=1.329;VQSLOD=-1.253e+01;VariantType=SNP;culprit=MQ;cytoBand=3p26.3;Func=intergenic;Gene=NONE,AY269186.1;GeneDetail=NONE,dist=5349;genomicSuperDups=0.95747,chr3:75279192;FATHMM_c=0.07045;FATHMM_nc=0.00765;CSQ=A|intergenic_variant|MODIFIER||||||||||||||||||SNV||||||||||||||||||||||||||||||||||||||||||||   0/0   .:7,0:7:21:0,21,224  GT:AB:AD:DP:GQ:PL 4.92E-05 0.00026455  NA 0  NA 0  NA 5  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA 4.92E-05
3  60082 NA G  A  NA NA NA NA NA NA NA NA NA NA intergenic_variant   NA ABCDEF   ABCDEF102   1  uncertain   PASS;ABHet=0.556;AC=2;AF=2.169e-04;AN=9220;BaseQRankSum=-2.890e-01;DP=58911;FS=0.000;GC=38.10;GQ_MEAN=37.95;GQ_STDDEV=12.17;InbreedingCoeff=-0.0017;MLEAC=2;MLEAF=2.169e-04;MQ=50.28;MQ0=0;MQRankSum=1.84;NCC=0;NDA=1;QD=6.89;ReadPosRankSum=0.361;SOR=1.329;VQSLOD=-1.253e+01;VariantType=SNP;culprit=MQ;cytoBand=3p26.3;Func=intergenic;Gene=NONE,AY269186.1;GeneDetail=NONE,dist=5349;genomicSuperDups=0.95747,chr3:75279192;FATHMM_c=0.07045;FATHMM_nc=0.00765;CSQ=A|intergenic_variant|MODIFIER||||||||||||||||||SNV||||||||||||||||||||||||||||||||||||||||||||   0/0   .:6,0:6:18:0,18,193  GT:AB:AD:DP:GQ:PL 4.92E-05 0.00026455  NA 0  NA 0  NA 5  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA 4.92E-05
```

**Merge All Chromosome Command for Subdirectory Input**
```
bash /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/Filtering_Wrapper_Script.sh --input_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_per_chr/subdir_input --input_prefix Transmission_Summary_Flat_chr --input_suffix _annotated_for_analysis.db --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_per_chr/merge_all_chr_output --input_type per_chr --flat_or_family flat --comb_and_clean merge_all_chr --consequence regulatory_region_variant
```
* Note: Same command as that for files within input_dir.

**Merge All Chromosome Output File**:
```
#bmap command: /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/filter_flat_annotated_transmission_summary.py --input_db /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_per_chr/subdir_input/1/Transmission_Summary_Flat_chr1_annotated_for_analysis.db --comb_and_clean no_comb_clean --input_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_per_chr/subdir_input --input_prefix Transmission_Summary_Flat_chr --input_suffix _annotated_for_analysis.db --output_dir /ifs/collabs/geschwind/bmap_scripts/annotate_transmission_summary/TS_filtering_examples/multiple_file_per_chr/merge_all_chr_output --input_type per_chr --flat_or_family flat --consequence regulatory_region_variant
Chr   Position Endpos   Ref   Alt   esp6500siv2_all   ExAC_ALL 1000g2015aug_all  cg46  CADD  CADD_Phred  Polyphen2_HDIV_score Polyphen2_HDIV_pred  Polyphen2_HVAR_score Polyphen2_HVAR_pred  consequence gene_symbol family   child_id is_aff   inheritance_type  Info  genotype genotype_subfields   subfield_format   max_control_AF HNP_AF   adjusted_hom_freq_PSP   adjusted_hom_freq_HNP   %PSP_samples_missing %HNP_samples_missing RDNV_category  GIAB_flag_count   HGNC_symbol HGNC_mapping   %RVIS_ESP_0.1% %RVIS_ExAC_0.01%  %RVIS_ExAC_0.05%popn %RVIS_ExAC_0.1%popn  constraint_score  pLI   pRecessive  pNull pHI   chrN:N-N pph2_prediction   pph2_prob   ARC_score   SSC_HNP_AF  adjusted_hom_freq_SSC_HNP  %SSC_HNP_samples_missing   gnomAD_AF
2  10205 NA T  G  NA NA NA NA NA NA NA NA NA NA regulatory_region_variant  NA ABCDEF   ABCDEF101   0  uncertain   PASS;ABHet=0.500;ABHom=NaN;AC=5;AF=1.132e-03;AN=4416;BaseQRankSum=0.804;DP=12674;FS=0.000;GC=61.90;GQ_MEAN=4.07;GQ_STDDEV=4.97;InbreedingCoeff=-0.1347;MLEAC=3;MLEAF=6.793e-04;MQ=59.67;MQ0=0;MQRankSum=0.804;NCC=2402;NDA=2;NEGATIVE_TRAIN_SITE;QD=34.24;ReadPosRankSum=0.804;SOR=0.538;VQSLOD=-1.159e+01;VariantType=SNP;culprit=DP;cytoBand=2p25.3;Func=intergenic;Gene=NONE,FAM110C;GeneDetail=NONE,dist=28609;FATHMM_c=0.21254;FATHMM_nc=0.36333;CSQ=G|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000675422|TF_binding_site||||||||||rs111214705|||SNV||||||||||||||||||||||||||||||||||||||||||||  0/0   .:3,0:3:0:0,0,40  GT:AB:AD:DP:GQ:PL 0  0.001088139 NA 0.001088139 NA 51.37566138 NA 2  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA
2  10205 NA T  G  NA NA NA NA NA NA NA NA NA NA regulatory_region_variant  NA ABCDEF   ABCDEF102   1  missing  PASS;ABHet=0.500;ABHom=NaN;AC=5;AF=1.132e-03;AN=4416;BaseQRankSum=0.804;DP=12674;FS=0.000;GC=61.90;GQ_MEAN=4.07;GQ_STDDEV=4.97;InbreedingCoeff=-0.1347;MLEAC=3;MLEAF=6.793e-04;MQ=59.67;MQ0=0;MQRankSum=0.804;NCC=2402;NDA=2;NEGATIVE_TRAIN_SITE;QD=34.24;ReadPosRankSum=0.804;SOR=0.538;VQSLOD=-1.159e+01;VariantType=SNP;culprit=DP;cytoBand=2p25.3;Func=intergenic;Gene=NONE,FAM110C;GeneDetail=NONE,dist=28609;FATHMM_c=0.21254;FATHMM_nc=0.36333;CSQ=G|regulatory_region_variant|MODIFIER|||RegulatoryFeature|ENSR00000675422|TF_binding_site||||||||||rs111214705|||SNV||||||||||||||||||||||||||||||||||||||||||||  ./.   .:0,0:0  GT:AB:AD:DP:GQ:PL 0  0.001088139 NA 0.001088139 NA 51.37566138 NA 2  NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA
```


## 4. Additional support scripts:
1. get_cohort_stats/get_flat_cohort_stats.py #Function? 
2. giab_dict_performance_test/ #A set of tests done to optimize performance for GIAB annotations?
3. line_counts #Old? Should be removed? 
4. add_gnomad_annotation_with_compatibility.py #Helpful script for annotating a flat file with gnomAD AFs. 
5. fats.py #Function? Used in filter_flat_annotated_transmission_summary.py
6. jobs_to_rerun.txt #Old? Should be removed?
7. qsub_patch_missing.sh #Function? 
8. qsub_patch_missing_family.sh #Function? 
9. qsub_tabix.sh #Makes .gz and .tbi VCF files to save space and improve speed for annotation/filtering.
10. summarize_annotated_transmission_summary_files_by_project.R #Summarizes flat and family aggregate transmission summary format per project 
11. divide_chr.py #Script to divide input db into per chromosome shards
12. divide_regions.py #Script to divide regions file into per chromosome files

## 5. Instructions for adding a new project:
1. Set project information in the script (currently starting on line #69): filter_annotated_transmission_summary_qsub.sh with a project name, directory_full_path, log_dir, and file_trunk. Add project name to Required arguments description for --project (currently line #6) and the Error line (currently line #95). 
2. Add project information (directory_full_path, file_trunk, and output_filename) to combine_and_clean.sh
3. Add project name to "Required_Parameters" on the google doc: https://docs.google.com/spreadsheets/d/1d28ojG4sLbMGYldkZrDMTaLf2rylM4_cCL7uvEEzZPo/edit?usp=sharing
4. Make a version of qsub_split_gz_ms2.sh, make project-specific edits, and submit this to split large transmission summary files. 
5. Make a version of qsub_shards_iHART.sh, make project-specific edits, and submit this to annotate transmission summary shards.
6. NOTE: If the split files now span a large genomic range (as is the case for a custom capture project) you may hit memory errors, in which you can request all 23G, you could add a parameter like -l h_vmem=23G to lines 38 and 46 in qsub_shards_NeuroReg.sh.

## 6. Hints for adding a new column:
1. If a new column is added it will go at the end of the columns (for consistency with other scripts)
2. Adding a new column into the annotated transmission summary file is relatively straightforward. Though you have to remember to add the code for adding a new column to both annotate_flat_transmission_summary.py and  annotate_family_aggregation_transmission_summary.py if desired.
3. Column names must be added to annotate_transmission_summary/fats.py in the static lists TransmissionSummaryLine.COLUMNS and FamilyAggregationTransmissionSummaryLine.FAMILY_AGGREGATION_COLUMNS if applicable (as well as TransmissionSummaryLine.NUMERIC_COLUMNS and FamilyAggregationTransmissionSummaryLine.NUMERIC_COLUMNS if the column should be treated as a numeric value). Otherwise you will have an error the first time you try to filter.

##### Documentation drafted by E.Ruzzo 2018-06-14, edited by S.Arteaga 2019-03-22.
